<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Omics Translation in AD: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="manifest" href="../site.webmanifest">
<link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav "><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="-logo" alt="" src="../assets/images/-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav " aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="" src="../assets/images/-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Omics Translation in AD
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Omics Translation in AD
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Omics Translation in AD
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress ">
    <div class="progress-bar " role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction to Omics Translation in AD</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="exploring-data-in-synapse.html">2. Synapse and AD Knowledge Portal</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="differential-expression-analysis.html">3. Differential expression analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="mouse-human-alignment.html">4. Mouse-human alignment of transcriptomic signatures</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-introduction"><p>Content from <a href="introduction.html">Introduction to Omics Translation in AD</a></p>
<hr>
<p>Last updated on 2024-07-25 |

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/episodes/introduction.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 10 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>This workshop will deliver hands-on training using the most recent
computational techniques and omics data resources to improve awareness
and utility of integrating multi-scale data from model systems and
humans to uncover the mechanistic drivers of Alzheimer’s Disease
(AD).</p>
<p>Despite considerable effort in drug development for AD, high rate of
failure in clinical trials emphasize the need for more effective
preclinical strategies, including targeted use of mouse models to study
heterogenous disease phenotypes. In this era of omics-driven research,
interspecies alignment of disease relevant molecular signatures relies
heavily on bioinformatics tools and field-specific data
infrastructures.</p>
<p>The goal of this workshop is to accelerate translational research in
AD by training interested scientists on how to develop robust
computational workflows for both hypothesis-driven and data-driven
research strategies. The AD-specific, bespoke lessons will use resources
from the AD Knowledge Portal (<a href="https://adknowledgeportal.synapse.org" class="external-link uri">https://adknowledgeportal.synapse.org</a>), a NIA designated
FAIR data repository that shares data from human and non-human studies
generated by multiple collaborative research programs focused on aging,
dementia, and AD.</p>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>Inline instructor notes can help inform instructors of timing
challenges associated with the lessons. They appear in the “Instructor
View”</p>
</div>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-exploring-data-in-synapse"><p>Content from <a href="exploring-data-in-synapse.html">Synapse and AD Knowledge Portal</a></p>
<hr>
<p>Last updated on 2024-07-25 |

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/episodes/exploring-data-in-synapse.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to work with Synapse R client?</li>
<li>How to work with data in AD Knowledge Portal?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain how to use Synapser Package.</li>
<li>Demonstrate how to locate data and metadata in the Portal.</li>
<li>Demonstrate how to download data from the Portal
programmatically.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="working-with-ad-portal-metadata">Working with AD Portal metadata<a class="anchor" aria-label="anchor" href="#working-with-ad-portal-metadata"></a>
<a class="anchor" aria-label="anchor" href="#working-with-ad-portal-metadata"></a>
</h2>
<hr class="half-width">
<p><strong>Metadata basics</strong></p>
<p>We have now downloaded several metadata files and an RNAseq counts
file from the portal. For our next exercises, we want to read those
files in as R data so we can work with them.</p>
<p>We can see from the download_table we got during the bulk download
step that we have five metadata files. Two of these should be the
individual and biospecimen files, and three of them are assay metadata
files.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">download_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">metadataType</span>, <span class="va">assay</span><span class="op">)</span></span></code></pre>
</div>
<p>We are only interested in RNAseq data, so we will only read in the
individual, biospecimen, and RNAseq assay metadata files.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># counts matrix</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="st">"../data/htseqcounts_5XFAD.txt"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"../data/Jax.IU.Pitt_5XFAD_individual_metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"../data/Jax.IU.Pitt_5XFAD_biospecimen_metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#assay metadata</span></span>
<span><span class="va">rna_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"../data/Jax.IU.Pitt_5XFAD_assay_RNAseq_metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s examine the data and metadata files a bit before we begin our
analyses.</p>
<p><strong>Counts data</strong></p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calling a tibble object will print the first ten rows in a nice tidy output; doing the same for a base R dataframe will print the whole thing until it runs out of memory. If you want to inspect a large dataframe, use `head(df)`</span></span>
<span><span class="va">counts</span></span></code></pre>
</div>
<p>The data file has a column of ENSEMBL gene ids and then a bunch of
columns with count data, where the column headers correspond to the
specimenIDs. These specimenIDs should all be in the RNAseq assay
metadata file, so let’s check.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># what does the RNAseq assay metadata look like?</span></span>
<span><span class="va">rna_meta</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># are all the column headers from the counts matrix (except the first "gene_id" column) in the assay metadata?</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Assay metadata</strong></p>
<p>The assay metadata contains information about how data was generated
on each sample in the assay. Each specimenID represents a unique sample.
We can use some tools from dplyr to explore the metadata.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># how many unique specimens were sequenced?</span></span>
<span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># were the samples all sequenced on the same platform?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">rna_meta</span>, <span class="va">platform</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># were there multiple sequencing batches reported?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">rna_meta</span>, <span class="va">sequencingBatch</span><span class="op">)</span> </span></code></pre>
</div>
<p><strong>Biospecimen metadata</strong></p>
<p>The biospecimen metadata contains specimen-level information,
including organ and tissue the specimen was taken from, how it was
prepared, etc. Each specimenID is mapped to an individualID.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all specimens from the RNAseq assay metadata file should be in the biospecimen file</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span> <span class="op">%in%</span> <span class="va">bio_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># but the biospecimen file also contains specimens from different assays</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">bio_meta</span><span class="op">$</span><span class="va">specimenID</span> <span class="op">%in%</span> <span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Individual metadata</strong></p>
<p>The individual metadata contains information about all the
individuals in the study, represented by unique individualIDs. For
humans, this includes information on age, sex, race, diagnosis, etc. For
MODEL-AD mouse models, the individual metadata has information on model
genotypes, stock numbers, diet, and more.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all individualIDs in the biospecimen file should be in the individual file</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">bio_meta</span><span class="op">$</span><span class="va">individualID</span> <span class="op">%in%</span> <span class="va">ind_meta</span><span class="op">$</span><span class="va">individualID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># which model genotypes are in this study?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">ind_meta</span>, <span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Joining metadata</strong></p>
<p>We use the three-file structure for our metadata because it allows us
to store metadata for each study in a tidy format. Every line in the
assay and biospecimen files represents a unique specimen, and every line
in the individual file represents a unique individual. This means the
files can be easily joined by specimenID and individualID to get all
levels of metadata that apply to a particular data file. We will use the
left_join() function from the dplyr package, and the %&gt;% operator
from the magrittr package. If you are unfamiliar with the pipe, think of
it as a shorthand for “take this (the preceding object) and do that (the
subsequent command)”. See here (<a href="https://magrittr.tidyverse.org/" class="external-link uri">https://magrittr.tidyverse.org/</a>) for more info on piping
in R.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># join all the rows in the assay metadata that have a match in the biospecimen metadata</span></span>
<span><span class="va">joined_meta</span> <span class="op">&lt;-</span> <span class="va">rna_meta</span> <span class="op">%&gt;%</span> <span class="co">#start with the rnaseq assay metadata</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">bio_meta</span>, by <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span>  <span class="co">#join rows from biospecimen that match specimenID </span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">ind_meta</span>, by <span class="op">=</span> <span class="st">"individualID"</span><span class="op">)</span> <span class="co"># join rows from individual that match individualID</span></span>
<span></span>
<span><span class="va">joined_meta</span></span></code></pre>
</div>
<p>We now have a very wide dataframe that contains all the available
metadata on each specimen in the RNAseq data from this study. This
procedure can be used to join the three types of metadata files for
every study in the AD Knowledge Portal, allowing you to filter
individuals and specimens as needed based on your analysis criteria!</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">lubridate</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># convert columns of strings to month-date-year format</span></span>
<span><span class="va">joined_meta_time</span> <span class="op">&lt;-</span> <span class="va">joined_meta</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>dateBirth <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateBirth</span><span class="op">)</span>, dateDeath <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateDeath</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="co"># create a new column that subtracts dateBirth from dateDeath in days, then divide by 30 to get months</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>timepoint <span class="op">=</span> <span class="fu">as.numeric</span><span class="op">(</span><span class="fu">difftime</span><span class="op">(</span><span class="va">dateDeath</span>, <span class="va">dateBirth</span>, units <span class="op">=</span><span class="st">"days"</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fl">30</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="co"># convert numeric ages to timepoint categories</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>timepoint <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span><span class="va">timepoint</span> <span class="op">&gt;</span> <span class="fl">10</span> <span class="op">~</span> <span class="st">"12 mo"</span>,</span>
<span>                               <span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">10</span> <span class="op">&amp;</span> <span class="va">timepoint</span> <span class="op">&gt;</span> <span class="fl">5</span> <span class="op">~</span> <span class="st">"6 mo"</span>,</span>
<span>                               <span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">5</span> <span class="op">~</span> <span class="st">"4 mo"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covars_5XFAD</span> <span class="op">&lt;-</span> <span class="va">joined_meta_time</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">individualID</span>, <span class="va">specimenID</span>, <span class="va">sex</span>, <span class="va">genotype</span>, <span class="va">timepoint</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">covars_5XFAD</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">covars_5XFAD</span><span class="op">$</span><span class="va">specimenID</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">covars_5XFAD</span><span class="op">)</span></span></code></pre>
</div>
<p>We will save joined_meta for the next lesson.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">covars_5XFAD</span>, file <span class="op">=</span> <span class="st">"../data/covars_5XFAD.rds"</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="single-specimen-files">Single Specimen files<a class="anchor" aria-label="anchor" href="#single-specimen-files"></a>
<a class="anchor" aria-label="anchor" href="#single-specimen-files"></a>
</h2>
<hr class="half-width">
<p>For files that contain data from a single specimen (e.g. raw
sequencing files, raw mass spectra, etc.), we can use the Synapse
annotations to associate these files with the appropriate metadata.</p>
<p>Excercise 3: Use Explore Data to find all RNAseq files from the
Jax.IU.Pitt_5XFAD study. If we filter for data where Study =
“Jax.IU.Pitt_5XFAD” and Assay = “rnaSeq” we will get a list of 148
files, including raw fastqs and processed counts data.</p>
<p>Synapse entity annotations We can use the function synGetAnnotations
to view the annotations associated with any file without downloading the
file.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the synID of a random fastq file from this list</span></span>
<span><span class="va">random_fastq</span> <span class="op">&lt;-</span> <span class="st">"syn22108503"</span></span>
<span></span>
<span><span class="co"># extract the annotations as a nested list</span></span>
<span><span class="va">fastq_annotations</span> <span class="op">&lt;-</span> <span class="fu">synGetAnnotations</span><span class="op">(</span><span class="va">random_fastq</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fastq_annotations</span></span></code></pre>
</div>
<p>The file annotations let us see which study the file is associated
with (Jax.IU.Pitt.5XFAD), which species it’s from (Mouse), which assay
generated the file (rnaSeq), and a whole bunch of other properties. Most
importantly, single-specimen files are annotated with with the
specimenID of the specimen in the file, and the individualID of the
individual that specimen was taken from. We can use these annotations to
link files to the rest of the metadata, including metadata that is not
in annotations. This is especially helpful for human studies, as
potentially identifying information like age, race, and diagnosis is not
included in file annotations.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># find records belonging to the individual this file maps to in our joined metadata</span></span>
<span><span class="va">joined_meta</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">individualID</span> <span class="op">==</span> <span class="va">fastq_annotations</span><span class="op">$</span><span class="va">individualID</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="annotations-during-bulk-download">Annotations during bulk download<a class="anchor" aria-label="anchor" href="#annotations-during-bulk-download"></a>
<a class="anchor" aria-label="anchor" href="#annotations-during-bulk-download"></a>
</h2>
<hr class="half-width">
<p>When bulk downloading many files, the best practice is to preserve
the download manifest that is generated which lists all the files, their
synIDs, and all their annotations. If using the Synapse R client, follow
the instructions in the Bulk download files section above.</p>
<p>If we use the “Programmatic Options” tab in the AD Portal download
menu to download all 148 rnaSeq files from the 5XFAD study, we would get
a table query that looks like this:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">synTableQuery</span><span class="op">(</span><span class="st">"SELECT * FROM syn11346063.37 WHERE ( ( \"study\" HAS ( 'Jax.IU.Pitt_5XFAD' ) ) AND ( \"assay\" HAS ( 'rnaSeq' ) ) )"</span><span class="op">)</span></span></code></pre>
</div>
<p>As we saw previously, this downloads a csv file with the results of
our AD Portal query. Opening that file lets us see which specimens are
associated with which files:</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations_table</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">query</span><span class="op">$</span><span class="va">filepath</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">annotations_table</span></span></code></pre>
</div>
<p>You could then use purrr::walk(download_table$id, ~synGet(.x,
downloadLocation = )) to walk through the column of synIDs and download
all 148 files. However, because these are large files, it might be
preferable to use the Python client or command line client for increased
speed.</p>
<p>Once you’ve downloaded all the files in the id column, you can link
those files to their annotations by the name column.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We'll use the "random fastq" that we got annotations for earlier</span></span>
<span><span class="co"># To avoid downloading the whole 3GB file, we'll use synGet with "downloadFile = FALSE" to get only the Synapse entity information, rather than the file. </span></span>
<span><span class="co"># If we downloaded the actual file, we could find it in the directory and search using the filename. Since we're just downloading the Synapse entity wrapper object, we'll use the file name listed in the object properties.</span></span>
<span></span>
<span><span class="va">fastq</span> <span class="op">&lt;-</span> <span class="fu">synGet</span><span class="op">(</span><span class="va">random_fastq</span>, downloadFile <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the annotations table to rows that match the fastq filename</span></span>
<span><span class="va">annotations_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="va">fastq</span><span class="op">$</span><span class="va">properties</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="multispecimen-files">Multispecimen files<a class="anchor" aria-label="anchor" href="#multispecimen-files"></a>
<a class="anchor" aria-label="anchor" href="#multispecimen-files"></a>
</h2>
<hr class="half-width">
<p>Multispecimen files in the AD Knowledge Portal are files that contain
data or information from more than one specimen. They are not annotated
with individualIDs or specimenIDs, since these files may contain numbers
of specimens that exceed the annotation limits. These files are usually
processed or summary data (gene counts, peptide quantifications, etc),
and are always annotated with isMultiSpecimen = TRUE.</p>
<p>If we look at the processed data files in the table of 5XFAD RNAseq
file annotations we just downloaded, we will see that it isMultiSpecimen
= TRUE, but individualID and specimenID are blank:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">fileFormat</span> <span class="op">==</span> <span class="st">"txt"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">individualID</span>, <span class="va">specimenID</span>, <span class="va">isMultiSpecimen</span><span class="op">)</span></span></code></pre>
</div>
<p>The multispecimen file should contain a row or column of specimenIDs
that correspond to the specimenIDs used in a study’s metadata, as we
have seen with the 5XFAD counts file.</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># In this example, we take a slice of the counts data to reduce computation, transpose it so that each row represents a single specimen, and then join it to the joined metadata by the specimenID</span></span>
<span><span class="va">counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">slice_head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">t</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span>rownames <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">joined_meta</span>, by <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>Use your Synapse login credentials to access the Portal.</li>
<li>Use Synapser package to download data from the Portal.</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessionInfo</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
[1] compiler_4.4.1  tools_4.4.1     yaml_2.3.9      knitr_1.48
[5] xfun_0.46       renv_1.0.7      evaluate_0.24.0</code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-differential-expression-analysis"><p>Content from <a href="differential-expression-analysis.html">Differential expression analysis</a></p>
<hr>
<p>Last updated on 2024-07-25 |

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/episodes/differential-expression-analysis.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What transcriptomic changes do we observe in mouse models carrying
AD-related mutations?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Read in a count matrix and metadata.</li>
<li>Understand the data from AD mouse models</li>
<li>Format the data for differential analysis</li>
<li>Perform differential analysis using DESeq2.</li>
<li>Pathway enrichment of differentially expressed genes</li>
<li>Save data for next lessons</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h1>
<p><strong>Reading Gene Expression Count matrix from Previous
Lesson</strong></p>
<p>In this lesson, we will use the raw counts matrix and metadata
downloaded in the previous lesson and will perform differential
expression analysis.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read.delim</span><span class="op">(</span><span class="st">"data/htseqcounts_5XFAD.txt"</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Reading Sample Metadata from Previous Lesson</strong></p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"data/covars_5XFAD.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s explore the data:</p>
<p>Let’s look at the top of the metadata.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">covars</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        individualID specimenID    sex         genotype timepoint
32043rh        32043    32043rh female    5XFAD_carrier     12 mo
32044rh        32044    32044rh   male 5XFAD_noncarrier     12 mo
32046rh        32046    32046rh   male 5XFAD_noncarrier     12 mo
32047rh        32047    32047rh   male 5XFAD_noncarrier     12 mo
32049rh        32049    32049rh female 5XFAD_noncarrier     12 mo
32057rh        32057    32057rh female 5XFAD_noncarrier     12 mo</code></pre>
</div>
<p>identify distinct groups using sample metadata</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">distinct</span><span class="op">(</span><span class="va">covars</span>, <span class="va">sex</span>, <span class="va">genotype</span>, <span class="va">timepoint</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in distinct(covars, sex, genotype, timepoint): could not find function "distinct"</code></pre>
</div>
<p>How many mice were used to produce this data?</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sex</span>,<span class="va">genotype</span>,<span class="va">timepoint</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in covars %&gt;% group_by(sex, genotype, timepoint) %&gt;% dplyr::count(): could not find function "%&gt;%"</code></pre>
</div>
<p>How many rows and columns are there in counts?</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 55489    73</code></pre>
</div>
<p>In the counts matrix, genes are in rows and samples are in columns.
Let’s look at the first few rows.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">counts</span>,n<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id 32043rh 32044rh 32046rh 32047rh 32048rh 32049rh 32050rh
1    ENSG00000080815   22554       0       0       0   16700       0       0
2    ENSG00000142192  344489       4       0       1  260935       6       8
3 ENSMUSG00000000001    5061    3483    3941    3088    2756    3067    2711
4 ENSMUSG00000000003       0       0       0       0       0       0       0
5 ENSMUSG00000000028     208     162     138     127      95     154     165
  32052rh 32053rh 32057rh 32059rh 32061rh 32062rh 32065rh 32067rh 32068rh
1   19748   14023       0   17062       0   15986      10       0   18584
2  337456  206851       1  264748       0  252248     172       4  300398
3    3334    3841    4068    3306    4076    3732    3940    4238    3257
4       0       0       0       0       0       0       0       0       0
5     124     103     164     116     108     134     204     239     148
  32070rh 32073rh 32074rh 32075rh 32078rh 32081rh 32088rh 32640rh 46105rh
1       1       0       0   22783   17029   16626   15573   12721       4
2       4       2       9  342655  280968  258597  243373  188818      19
3    3351    3449    4654    4844    3132    3334    3639    3355    4191
4       0       0       0       0       0       0       0       0       0
5     159     167     157     211     162     149     160     103     158
  46106rh 46107rh 46108rh 46109rh 46110rh 46111rh 46112rh 46113rh 46115rh
1       0       0       0       0       0   17931       0   19087       0
2       0       0       1       5       1  293409       8  273704       1
3    3058    4265    3248    3638    3747    3971    3192    3805    3753
4       0       0       0       0       0       0       0       0       0
5     167     199     113     168     175     203     158     108     110
  46121rh 46131rh 46132rh 46134rh 46138rh 46141rh 46142rh 46862rh 46863rh
1       0       0   12703   18833       0   18702   17666       0   14834
2       0       1  187975  285048       0  284499  250600       0  218494
3    4134    3059    3116    3853    3682    2844    3466    3442    3300
4       0       0       0       0       0       0       0       0       0
5     179     137     145     183     171     138      88     154     157
  46865rh 46866rh 46867rh 46868rh 46871rh 46872rh 46873rh 46874rh 46875rh
1   10546   10830   10316   10638   15248       0       0   11608   11561
2  169516  152769  151732  190150  229063       6       1  165941  171303
3    3242    3872    3656    3739    3473    3154    5510    3657    4121
4       0       0       0       0       0       0       0       0       0
5     131     152     152     155     140      80     240     148     112
  46876rh 46877rh 46878rh 46879rh 46881rh 46882rh 46883rh 46884rh 46885rh
1       0       0   12683   15613       0   14084   20753       0       0
2       0       2  183058  216122       0  199448  306081       0       5
3    3422    3829    3996    4324    2592    2606    4600    2913    3614
4       0       0       0       0       0       0       0       0       0
5     147     166     169     215     115     101     174     127     151
  46886rh 46887rh 46888rh 46889rh 46890rh 46891rh 46892rh 46893rh 46895rh
1   16639   16072       0   16680   13367       0   25119      92       0
2  242543  258061       0  235530  196721       0  371037    1116       0
3    3294    3719    3899    4173    4008    3037    5967    3459    4262
4       0       0       0       0       0       0       0       0       0
5     139     128     210     127     156     116     260     161     189
  46896rh 46897rh
1   15934       0
2  235343       6
3    3923    3486
4       0       0
5     179     117</code></pre>
</div>
<p>As you can see, the gene ids are ENSEMBL IDs. There is some risk that
these may not be unique. Let’s check whether any of the gene symbols are
duplicated. We will sum the number of duplicated gene symbols.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="fu">duplicated</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 0</code></pre>
</div>
<p>The sum equals zero, so there are no duplicated gene symbols, which
is good. Similarly, samples should be unique. Once again, let’s verify
this:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="fu">duplicated</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 0</code></pre>
</div>
<p><strong>Formatting the count matrix</strong></p>
<p>Now, as we see that gene_id is in first column of count matrix, but
we will need only count data in matrix, so we will change the gene_id
column to rownames. Converting the gene_id as rownames of count
matrix</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">column_to_rownames</span><span class="op">(</span><span class="va">.</span>,var<span class="op">=</span><span class="st">"gene_id"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in counts %&gt;% column_to_rownames(., var = "gene_id") %&gt;% as.data.frame(): could not find function "%&gt;%"</code></pre>
</div>
<p>let’s confirm if change is done correctly</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">counts</span>, n<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id 32043rh 32044rh 32046rh 32047rh 32048rh 32049rh 32050rh
1    ENSG00000080815   22554       0       0       0   16700       0       0
2    ENSG00000142192  344489       4       0       1  260935       6       8
3 ENSMUSG00000000001    5061    3483    3941    3088    2756    3067    2711
4 ENSMUSG00000000003       0       0       0       0       0       0       0
5 ENSMUSG00000000028     208     162     138     127      95     154     165
  32052rh 32053rh 32057rh 32059rh 32061rh 32062rh 32065rh 32067rh 32068rh
1   19748   14023       0   17062       0   15986      10       0   18584
2  337456  206851       1  264748       0  252248     172       4  300398
3    3334    3841    4068    3306    4076    3732    3940    4238    3257
4       0       0       0       0       0       0       0       0       0
5     124     103     164     116     108     134     204     239     148
  32070rh 32073rh 32074rh 32075rh 32078rh 32081rh 32088rh 32640rh 46105rh
1       1       0       0   22783   17029   16626   15573   12721       4
2       4       2       9  342655  280968  258597  243373  188818      19
3    3351    3449    4654    4844    3132    3334    3639    3355    4191
4       0       0       0       0       0       0       0       0       0
5     159     167     157     211     162     149     160     103     158
  46106rh 46107rh 46108rh 46109rh 46110rh 46111rh 46112rh 46113rh 46115rh
1       0       0       0       0       0   17931       0   19087       0
2       0       0       1       5       1  293409       8  273704       1
3    3058    4265    3248    3638    3747    3971    3192    3805    3753
4       0       0       0       0       0       0       0       0       0
5     167     199     113     168     175     203     158     108     110
  46121rh 46131rh 46132rh 46134rh 46138rh 46141rh 46142rh 46862rh 46863rh
1       0       0   12703   18833       0   18702   17666       0   14834
2       0       1  187975  285048       0  284499  250600       0  218494
3    4134    3059    3116    3853    3682    2844    3466    3442    3300
4       0       0       0       0       0       0       0       0       0
5     179     137     145     183     171     138      88     154     157
  46865rh 46866rh 46867rh 46868rh 46871rh 46872rh 46873rh 46874rh 46875rh
1   10546   10830   10316   10638   15248       0       0   11608   11561
2  169516  152769  151732  190150  229063       6       1  165941  171303
3    3242    3872    3656    3739    3473    3154    5510    3657    4121
4       0       0       0       0       0       0       0       0       0
5     131     152     152     155     140      80     240     148     112
  46876rh 46877rh 46878rh 46879rh 46881rh 46882rh 46883rh 46884rh 46885rh
1       0       0   12683   15613       0   14084   20753       0       0
2       0       2  183058  216122       0  199448  306081       0       5
3    3422    3829    3996    4324    2592    2606    4600    2913    3614
4       0       0       0       0       0       0       0       0       0
5     147     166     169     215     115     101     174     127     151
  46886rh 46887rh 46888rh 46889rh 46890rh 46891rh 46892rh 46893rh 46895rh
1   16639   16072       0   16680   13367       0   25119      92       0
2  242543  258061       0  235530  196721       0  371037    1116       0
3    3294    3719    3899    4173    4008    3037    5967    3459    4262
4       0       0       0       0       0       0       0       0       0
5     139     128     210     127     156     116     260     161     189
  46896rh 46897rh
1   15934       0
2  235343       6
3    3923    3486
4       0       0
5     179     117</code></pre>
</div>
<p>As you can see from count table there are some genes that start with
“ENSG” and others start with “ENSMUSG”. “ENSG” referes to human gene
ENSEMBL id and “ENSMUSG” refer to mouse ENSEMBL id. Let’s check how many
gene_ids are NOT from the mouse genome by searching for the string “MUS”
(as in Mus musculus) in the rownames of count matrix</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"MUS"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in counts[, 1:6] %&gt;% filter(!str_detect(rownames(.), "MUS")): could not find function "%&gt;%"</code></pre>
</div>
<p>Ok, so we see there are two human genes in out count matrix. Why?
What genes are they?</p>
<p>Briefly, 5xFAD mouse strain harbors two human transgenes APP
(“ENSG00000142192”) and PSEN1 (“ENSG00000080815”) and inserted into exon
2 of the mouse Thy1 gene. To validate 5XFAD strain and capture
expression of human transgene APP and PS1, a custom mouse genomic
sequence was created and we quantified expression of human as well as
mouse App (“ENSMUSG00000022892”) and Psen1 (“ENSMUSG00000019969”) genes
by our MODEL-AD RNA-Seq pipeline.</p>
<p><strong>Validation of 5xFAD mouse strain</strong></p>
<p>First we convert the dataframe to longer format and join our
covariates by MouseID</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_tpose</span> <span class="op">&lt;-</span> <span class="va">counts</span>  <span class="op">%&gt;%</span></span>
<span>                <span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>,var<span class="op">=</span><span class="st">"gene_id"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                <span class="fu">filter</span><span class="op">(</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span>,<span class="st">"ENSMUSG00000019969"</span>,<span class="st">"ENSG00000142192"</span>,<span class="st">"ENSMUSG00000022892"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">.</span>,cols <span class="op">=</span> <span class="op">-</span><span class="st">"gene_id"</span>,names_to <span class="op">=</span> <span class="st">"specimenID"</span>,values_to<span class="op">=</span><span class="st">"counts"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu">left_join</span><span class="op">(</span><span class="va">covars</span>, by<span class="op">=</span><span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in counts %&gt;% rownames_to_column(., var = "gene_id") %&gt;% filter(gene_id %in% : could not find function "%&gt;%"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">count_tpose</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'count_tpose' not found</code></pre>
</div>
<p>Rename the APP and PSEN1 genes to specify whether mouse or human.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#make the age column a factor and re-order the levels</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">timepoint</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">timepoint</span>,levels<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"4 mo"</span>,<span class="st">"6 mo"</span>,<span class="st">"12 mo"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'count_tpose' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># rename the gene id to gene symbol</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSG00000142192"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human APP"</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'count_tpose' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSG00000080815"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human PSEN1"</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'count_tpose' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mouse App"</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'count_tpose' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mouse Psen1"</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error: object 'count_tpose' not found</code></pre>
</div>
<p>Visualize orthologous genes.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Create simple box plots showing normalized counts by genotype and time point, faceted by sex.</span></span>
<span><span class="va">count_tpose</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">timepoint</span>, y<span class="op">=</span><span class="va">counts</span>, color<span class="op">=</span><span class="va">genotype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>position<span class="op">=</span><span class="fu">position_jitterdodge</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">+</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">+</span><span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in count_tpose %&gt;% ggplot(aes(x = timepoint, y = counts, color = genotype)): could not find function "%&gt;%"</code></pre>
</div>
<p>You will notice expression of Human APP is higher in 5XFAD carriers
but lower in non-carriers. However mouse App expressed in both 5XFAD
carrier and non-carrier.</p>
<p>We are going to sum the counts from both ortholgous genes (human APP
and mouse App; human PSEN1 and mouse Psen1) and save summed expression
as expression of mouse genes, respectively to match with gene names in
control mice.</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#merge mouse and human APP gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>,<span class="op">]</span> <span class="op">+</span> <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000142192"</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in FUN(left, right): non-numeric argument to binary operator</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000142192"</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#merge mouse and human PS1 gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>,<span class="op">]</span> <span class="op">+</span> <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000080815"</span>,<span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in FUN(left, right): non-numeric argument to binary operator</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span><span class="op">)</span>,<span class="op">]</span></span></code></pre>
</div>
<p>Let’s verify if expression of both human genes have been merged or
not:</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"MUS"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in counts[, 1:6] %&gt;% filter(!str_detect(rownames(.), "MUS")): could not find function "%&gt;%"</code></pre>
</div>
<p>What proportion of genes have zero counts in all samples?</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sums</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>                        sums    <span class="op">=</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">rowSums</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in base::rowSums(x, na.rm = na.rm, dims = dims, ...): 'x' must be numeric</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">gene_sums</span><span class="op">$</span><span class="va">sums</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'gene_sums' not found</code></pre>
</div>
<p>We can see that 9691 (17%) genes have no reads at all associated with
them. In the next lesson, we will remove genes that have no counts in
any samples.</p>
<div class="section level2">
<h2 id="differential-analysis-using-deseq2">Differential Analysis using DESeq2<a class="anchor" aria-label="anchor" href="#differential-analysis-using-deseq2"></a>
</h2>
<p>Now, after exploring and formatting the data, We will look for
differential expression between the control and 5xFAD mice at different
ages for both sexes. The differentially expressed genes (DEGs) can
inform our understanding of how the 5XFAD mutation affect the biological
processes.</p>
<p>DESeq2 analysis consist of multiple steps. We are going to briefly
understand some of the important steps using subset of data and then we
will perform differential analysis on whole dataset.</p>
<p>First, order the data (so counts and metadata specimenID orders
match) and save as another variable name</p>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rawdata</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>,<span class="fu">sort</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">covars</span><span class="op">[</span><span class="fu">sort</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">covars</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span></span></code></pre>
</div>
<p>subset the counts matrix and sample metadata to include only 12 month
old male mice. You can amend the code to compare wild type and 5XFAD
mice from either sex, at any time point.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.12M.Male</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">sex</span><span class="op">==</span><span class="st">"male"</span> <span class="op">&amp;</span> <span class="va">metadata</span><span class="op">$</span><span class="va">timepoint</span><span class="op">==</span><span class="st">'12 mo'</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">meta.12M.Male</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        individualID specimenID  sex         genotype timepoint
32044rh        32044    32044rh male 5XFAD_noncarrier     12 mo
32046rh        32046    32046rh male 5XFAD_noncarrier     12 mo
32047rh        32047    32047rh male 5XFAD_noncarrier     12 mo
32053rh        32053    32053rh male    5XFAD_carrier     12 mo
32059rh        32059    32059rh male    5XFAD_carrier     12 mo
32061rh        32061    32061rh male 5XFAD_noncarrier     12 mo
32062rh        32062    32062rh male    5XFAD_carrier     12 mo
32073rh        32073    32073rh male 5XFAD_noncarrier     12 mo
32074rh        32074    32074rh male 5XFAD_noncarrier     12 mo
32075rh        32075    32075rh male    5XFAD_carrier     12 mo
32088rh        32088    32088rh male    5XFAD_carrier     12 mo
32640rh        32640    32640rh male    5XFAD_carrier     12 mo</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">as.matrix</span><span class="op">(</span><span class="va">rawdata</span><span class="op">[</span>,<span class="fu">colnames</span><span class="op">(</span><span class="va">rawdata</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">meta.12M.Male</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "32044rh" "32046rh" "32047rh" "32053rh" "32059rh" "32061rh" "32062rh"
 [8] "32073rh" "32074rh" "32075rh" "32088rh" "32640rh"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rownames</span><span class="op">(</span><span class="va">meta.12M.Male</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "32044rh" "32046rh" "32047rh" "32053rh" "32059rh" "32061rh" "32062rh"
 [8] "32073rh" "32074rh" "32075rh" "32088rh" "32640rh"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">match</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">dat</span><span class="op">)</span>,<span class="fu">rownames</span><span class="op">(</span><span class="va">meta.12M.Male</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1]  1  2  3  4  5  6  7  8  9 10 11 12</code></pre>
</div>
<p>Next, we build the DESeqDataSet using the following function:</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData<span class="op">=</span><span class="va">dat</span>, </span>
<span>                                   colData<span class="op">=</span><span class="va">meta.12M.Male</span>, </span>
<span>                                   design <span class="op">=</span> <span class="op">~</span><span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<p><strong>Pre-filtering</strong></p>
<p>While it is not necessary to pre-filter low count genes before
running the DESeq2 functions, there are two reasons which make
pre-filtering useful: by removing rows in which there are very few
reads, we reduce the memory size of the dds data object, and we increase
the speed of the transformation and testing functions within DESeq2. It
can also improve visualizations, as features with no information for
differential expression are not plotted.</p>
<p>Here we perform a minimal pre-filtering to keep only rows that have
at least 10 reads total.</p>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<p><strong>Reference level</strong></p>
<p>By default, R will choose a reference level for factors based on
alphabetical order. Then, if you never tell the DESeq2 functions which
level you want to compare against (e.g. which level represents the
control group), the comparisons will be based on the alphabetical order
of the levels.</p>
<p>specifying the reference-level to 5XFAD_noncarrier:</p>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span>,ref<span class="op">=</span><span class="st">"5XFAD_noncarrier"</span><span class="op">)</span>  </span></code></pre>
</div>
<p>Run the standard differential expression analysis steps that is
wrapped into a single function, DESeq.</p>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>,parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>Results tables are generated using the function results, which
extracts a results table with log2 fold changes, p values and adjusted p
values. By default the argument alpha is set to 0.1. If the adjusted p
value cutoff will be a value other than 0.1, alpha should be set to that
value:</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span>,alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>  <span class="co"># setting 0.05 as significant threshold</span></span>
<span><span class="va">res</span></span></code></pre>
</div>
<p>We can order our results table by the smallest p value:</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resOrdered</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">resOrdered</span>,n<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>we can summarize some basic tallies using the summary function.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
</div>
<p>How many adjusted p-values were less than 0.05?</p>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>How many adjusted p-values were less than 0.1?</p>
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.1</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Function to convert ensembleIDs to common gene
names</strong></p>
<p>We’ll use a package to translate mouse ENSEMBL IDS to gene names. Run
this function and they will be called up when assembling results from
the differential expression analysis</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map_function.df</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">inputtype</span>, <span class="va">outputtype</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">mapIds</span><span class="op">(</span></span>
<span>    <span class="va">org.Mm.eg.db</span>,</span>
<span>    keys <span class="op">=</span> <span class="fu">row.names</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>    column <span class="op">=</span> <span class="va">outputtype</span>,</span>
<span>    keytype <span class="op">=</span> <span class="va">inputtype</span>,</span>
<span>    multiVals <span class="op">=</span> <span class="st">"first"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p><strong>Generating Results table</strong></p>
<p>Here we will call the function to get the ‘symbol’ names of the genes
incorporated into the results table, along with the columns we are most
interested in</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">All_res</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"SYMBOL"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>    <span class="co">##run map_function to add symbol of gene corresponding to ENSEBL ID</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>EntrezGene <span class="op">=</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>  <span class="co">##run map_function to add Entrez ID of gene corresponding to ENSEBL ID</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="st">"symbol"</span>, <span class="st">"EntrezGene"</span>,<span class="st">"baseMean"</span>, <span class="st">"log2FoldChange"</span>, <span class="st">"lfcSE"</span>, <span class="st">"stat"</span>, <span class="st">"pvalue"</span>, <span class="st">"padj"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">All_res</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Extracting genes that are significantly
expressed</strong></p>
<p>Let’s subset all the genes with a pvalue &lt; 0.05</p>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dseq_res</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">All_res</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">All_res</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<p>Wow! We have a lot of genes with apparently very strong statistically
significant differences between the control and 5xFAD carrier.</p>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">dseq_res</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">dseq_res</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="exploring-and-exporting-results">Exploring and exporting results<a class="anchor" aria-label="anchor" href="#exploring-and-exporting-results"></a>
</h2>
<p><strong>Exporting results to CSV files</strong></p>
<p>we can save results file into a csv file like this:</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">write.csv</span><span class="op">(</span><span class="va">All_res</span>,file<span class="op">=</span><span class="st">"results/All_5xFAD_12months_male.csv"</span><span class="op">)</span></span>
<span><span class="fu">write.csv</span><span class="op">(</span><span class="va">dseq_res</span>,file<span class="op">=</span><span class="st">"results/DEG_5xFAD_12months_male.csv"</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Volcano plot</strong></p>
<p>We can visualize the differential expression results using Volcano
plot function from EnhancedVolcano package. For the most basic volcano
plot, only a single data-frame, data-matrix, or tibble of test results
is required, containing point labels, log2FC, and adjusted or unadjusted
P values. The default cut-off for log2FC is &gt;|2|; the default cut-off
for P value is 10e-6.</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EnhancedVolcano</span><span class="op">(</span><span class="va">All_res</span>,</span>
<span>                                   lab <span class="op">=</span> <span class="op">(</span><span class="va">All_res</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span>,</span>
<span>                                   x <span class="op">=</span> <span class="st">'log2FoldChange'</span>,</span>
<span>                                   y <span class="op">=</span> <span class="st">'padj'</span>,legendPosition <span class="op">=</span> <span class="st">'none'</span>,</span>
<span>                                   title <span class="op">=</span> <span class="st">'Volcano plot:Differential Expression Results'</span>,</span>
<span>                                   subtitle <span class="op">=</span> <span class="st">''</span>,</span>
<span>                                   FCcutoff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                                   pCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                   xlim <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>You can see that some top significantly expressed are
immune/inflammation-related genes such as Ctsd, C4b, Csf1 etc. These
genes are upregulated in the 5XFAD strain.</p>
<p><strong>Principal component plot of the samples</strong></p>
<p>Principal component analysis is a dimension reduction technique that
reduces the dimensionality of these large matrixes into a linear
coordinate system, so that we can more easily visualize what factors are
contributing the most to variation in the dataset by graphing the
principal components.</p>
<div class="codewrapper sourceCode" id="cb76">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData<span class="op">=</span><span class="fu">as.matrix</span><span class="op">(</span><span class="va">rawdata</span><span class="op">)</span>, colData<span class="op">=</span><span class="va">metadata</span>, design<span class="op">=</span> <span class="op">~</span> <span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb77">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">1</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span>, <span class="op">]</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>,parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb78">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vsd</span> <span class="op">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span><span class="op">(</span><span class="va">dds</span>, blind<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plotPCA</span><span class="op">(</span><span class="va">vsd</span>, intgroup<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"genotype"</span>, <span class="st">"sex"</span>,<span class="st">"timepoint"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that clustering is occurring, though it’s kind of hard to
see exactly how they are clustering in this visualization.</p>
<p>It is also possible to customize the PCA plot using the ggplot
function.</p>
<div class="codewrapper sourceCode" id="cb79">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcaData</span> <span class="op">&lt;-</span> <span class="fu">plotPCA</span><span class="op">(</span><span class="va">vsd</span>, intgroup<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"genotype"</span>, <span class="st">"sex"</span>,<span class="st">"timepoint"</span><span class="op">)</span>, returnData<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">percentVar</span> <span class="op">&lt;-</span> <span class="fu">round</span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu">attr</span><span class="op">(</span><span class="va">pcaData</span>, <span class="st">"percentVar"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">pcaData</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">PC1</span>, <span class="va">PC2</span>,color<span class="op">=</span><span class="va">genotype</span>, shape<span class="op">=</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_text</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">timepoint</span><span class="op">)</span>,hjust<span class="op">=</span><span class="fl">0.5</span>, vjust<span class="op">=</span><span class="fl">2</span>,size <span class="op">=</span><span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x<span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"PC1: "</span>,<span class="va">percentVar</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,<span class="st">"% variance"</span><span class="op">)</span>, y<span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"PC2: "</span>,<span class="va">percentVar</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,<span class="st">"% variance"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>PCA identified genotype and sex being a major source of variation in
between 5XFAD and WT mice. Female and male samples from the 5XFAD
carriers clustered distinctly at all ages, suggesting the presence of
sex-biased molecular changes in animals.</p>
<p><strong>Function for Differential analysis using DESeq2</strong></p>
<p>Finally, we can build a function for differential analysis that
consists of all above discussed steps. It will require to input sorted
raw count matrix, sample metadata and define the reference group.</p>
<div class="codewrapper sourceCode" id="cb80">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">rawdata</span>, <span class="va">meta</span>,</span>
<span>                   <span class="va">include.batch</span> <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                   <span class="va">ref</span> <span class="op">=</span> <span class="va">ref</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">dseq_res</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">All_res</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">include.batch</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">cat</span><span class="op">(</span><span class="st">"Including batch as covariate\n"</span><span class="op">)</span></span>
<span>    <span class="va">design_formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Batch</span> <span class="op">+</span> <span class="va">genotype</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="kw">else</span><span class="op">{</span></span>
<span>    <span class="va">design_formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">genotype</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="va">dat2</span> <span class="op">&lt;-</span> <span class="fu">as.matrix</span><span class="op">(</span><span class="va">rawdata</span><span class="op">[</span>, <span class="fu">colnames</span><span class="op">(</span><span class="va">rawdata</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="va">ddsHTSeq</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="va">dat2</span>,</span>
<span>                           colData <span class="op">=</span> <span class="va">meta</span>,</span>
<span>                           design <span class="op">=</span> <span class="va">design_formula</span><span class="op">)</span></span>
<span>  <span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span>, <span class="op">]</span></span>
<span>  <span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span>, ref <span class="op">=</span> <span class="va">ref</span><span class="op">)</span></span>
<span>  <span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span>  <span class="co">#summary(res)</span></span>
<span>  </span>
<span>  <span class="va">res</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">res</span><span class="op">$</span><span class="va">EntrezGene</span> <span class="op">&lt;-</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">All_res</span> <span class="op">&lt;&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">res</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span>, <span class="st">"EntrezGene"</span>,<span class="st">"baseMean"</span>, <span class="st">"log2FoldChange"</span>, <span class="st">"lfcSE"</span>, <span class="st">"stat"</span>, <span class="st">"pvalue"</span>, <span class="st">"padj"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>Let’s use this function to analyze all groups present in our
data.</p>
<p><strong>Differential Analysis of all groups</strong></p>
<p>First, we add a Group column to our metadata table that will combine
all variables of interest (genotype, sex, and timepoint) for each
sample.</p>
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">genotype</span>,<span class="st">"-"</span>,<span class="va">metadata</span><span class="op">$</span><span class="va">sex</span>,<span class="st">"-"</span>,<span class="va">metadata</span><span class="op">$</span><span class="va">timepoint</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">unique</span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "5XFAD_carrier-female-12 mo"    "5XFAD_noncarrier-male-12 mo"
 [3] "5XFAD_noncarrier-female-12 mo" "5XFAD_carrier-male-12 mo"
 [5] "5XFAD_noncarrier-female-6 mo"  "5XFAD_noncarrier-male-6 mo"
 [7] "5XFAD_carrier-female-6 mo"     "5XFAD_noncarrier-female-4 mo"
 [9] "5XFAD_carrier-female-4 mo"     "5XFAD_carrier-male-6 mo"
[11] "5XFAD_carrier-male-4 mo"       "5XFAD_noncarrier-male-4 mo"   </code></pre>
</div>
<p>Next, we create a comparison table that has all cases and controls
that we would like to compare with each other. Here I have made
comparison groups for age and sex-matched 5xFAD carriers vs
5xFAD_noncarriers, with carriers as the cases and noncarriers as the
controls:</p>
<div class="codewrapper sourceCode" id="cb83">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span>  <span class="fu">data.frame</span><span class="op">(</span>control<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"5XFAD_noncarrier-male-4 mo"</span>, <span class="st">"5XFAD_noncarrier-female-4 mo"</span>, <span class="st">"5XFAD_noncarrier-male-6 mo"</span>, </span>
<span>                                     <span class="st">"5XFAD_noncarrier-female-6 mo"</span>,<span class="st">"5XFAD_noncarrier-male-12 mo"</span>, <span class="st">"5XFAD_noncarrier-female-12 mo"</span><span class="op">)</span>, </span>
<span>                           case<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"5XFAD_carrier-male-4 mo"</span>, <span class="st">"5XFAD_carrier-female-4 mo"</span>, <span class="st">"5XFAD_carrier-male-6 mo"</span>, </span>
<span>                                  <span class="st">"5XFAD_carrier-female-6 mo"</span>,<span class="st">"5XFAD_carrier-male-12 mo"</span>, <span class="st">"5XFAD_carrier-female-12 mo"</span><span class="op">)</span></span>
<span>                           <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb84">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                        control                       case
1    5XFAD_noncarrier-male-4 mo    5XFAD_carrier-male-4 mo
2  5XFAD_noncarrier-female-4 mo  5XFAD_carrier-female-4 mo
3    5XFAD_noncarrier-male-6 mo    5XFAD_carrier-male-6 mo
4  5XFAD_noncarrier-female-6 mo  5XFAD_carrier-female-6 mo
5   5XFAD_noncarrier-male-12 mo   5XFAD_carrier-male-12 mo
6 5XFAD_noncarrier-female-12 mo 5XFAD_carrier-female-12 mo</code></pre>
</div>
<p>Finally, we implement our DEG function on each case/control
comparison of interest and store the result table in a list and data
frame:</p>
<div class="codewrapper sourceCode" id="cb86">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># initiate an empty list and data frame to save results</span></span>
<span><span class="va">DE_5xFAD.list</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">DE_5xFAD.df</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu">nrow</span><span class="op">(</span><span class="va">comparisons</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">{</span></span>
<span>    <span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="va">metadata</span><span class="op">$</span><span class="va">Group</span> <span class="op">%in%</span> <span class="va">comparisons</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span>,<span class="op">]</span></span>
<span>    <span class="fu">DEG</span><span class="op">(</span><span class="va">rawdata</span>,<span class="va">meta</span>,ref <span class="op">=</span> <span class="st">"5XFAD_noncarrier"</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="co">#append results in data frame</span></span>
<span>    <span class="va">DE_5xFAD.df</span> <span class="op">&lt;-</span> <span class="fu">rbind</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>,<span class="va">All_res</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model<span class="op">=</span><span class="st">"5xFAD"</span>,sex<span class="op">=</span><span class="fu">unique</span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span>,age<span class="op">=</span><span class="fu">unique</span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">timepoint</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="co">#append results in list</span></span>
<span>    <span class="va">DE_5xFAD.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">All_res</span></span>
<span>    <span class="fu">names</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">comparisons</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>Let’s explore the result stored in our list:</p>
<div class="codewrapper sourceCode" id="cb87">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">)</span></span></code></pre>
</div>
<p>We can easily extract the result table for any group of interest by
using $ and name of group. Let’s check top few rows from
5XFAD_carrier-male-4 mo group:</p>
<div class="codewrapper sourceCode" id="cb88">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-male-4 mo`</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s check the result stored as dataframe:</p>
<div class="codewrapper sourceCode" id="cb89">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">DE_5xFAD.df</span><span class="op">)</span></span></code></pre>
</div>
<p>Check if result is present for all ages:</p>
<div class="codewrapper sourceCode" id="cb90">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="op">(</span><span class="va">DE_5xFAD.df</span><span class="op">$</span><span class="va">age</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Check if result is present for both sexes:</p>
<div class="codewrapper sourceCode" id="cb91">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="op">(</span><span class="va">DE_5xFAD.df</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Check number of genes in each group:</p>
<div class="codewrapper sourceCode" id="cb92">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>,<span class="va">model</span>,<span class="va">sex</span>,<span class="va">age</span><span class="op">)</span></span></code></pre>
</div>
<p>Check number of genes significantly differentially expressed in all
cases compared to age and sex-matched controls:</p>
<div class="codewrapper sourceCode" id="cb93">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">degs.up</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">DE_5xFAD.list</span>, <span class="op">~</span><span class="fu">length</span><span class="op">(</span><span class="fu">which</span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">padj</span><span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">.x</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">degs.down</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">DE_5xFAD.list</span>, <span class="op">~</span><span class="fu">length</span><span class="op">(</span><span class="fu">which</span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">padj</span><span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">.x</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>Cases<span class="op">=</span><span class="fu">names</span><span class="op">(</span><span class="va">degs.up</span><span class="op">)</span>, Up_DEGs.pval.05<span class="op">=</span><span class="fu">as.vector</span><span class="op">(</span><span class="fu">unlist</span><span class="op">(</span><span class="va">degs.up</span><span class="op">)</span><span class="op">)</span>,Down_DEGs.pval.05<span class="op">=</span><span class="fu">as.vector</span><span class="op">(</span><span class="fu">unlist</span><span class="op">(</span><span class="va">degs.down</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu">kable</span><span class="op">(</span><span class="va">deg</span><span class="op">)</span></span></code></pre>
</div>
<p>Interestingly, in females more genes are differentially expressed at
all age groups, and more genes are differentially expressed the older
the mice get in both sexes.</p>
</div>
<div class="section level2">
<h2 id="pathway-enrichment">Pathway Enrichment<a class="anchor" aria-label="anchor" href="#pathway-enrichment"></a>
</h2>
<p>We may wish to look for enrichment of biological pathways in a list
of differentially expressed genes. Here we will test for enrichment of
KEGG pathways using using enrichKEGG function in clusterProfiler
package.</p>
<div class="codewrapper sourceCode" id="cb94">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span>FAD_M_4<span class="op">=</span><span class="fu">subset</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-male-4 mo`</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-male-4 mo`</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">EntrezGene</span><span class="op">)</span>,</span>
<span>            FAD_F_4<span class="op">=</span><span class="fu">subset</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-female-4 mo`</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-female-4 mo`</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">EntrezGene</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## perform enrichment analysis</span></span>
<span><span class="va">enrich_pathway</span> <span class="op">&lt;-</span> <span class="fu">compareCluster</span><span class="op">(</span><span class="va">dat</span>,</span>
<span>                                 fun <span class="op">=</span> <span class="st">"enrichKEGG"</span>,</span>
<span>                                  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                  organism <span class="op">=</span> <span class="st">"mmu"</span></span>
<span>                                 <span class="op">)</span></span>
<span></span>
<span><span class="va">enrich_pathway</span><span class="op">@</span><span class="va">compareClusterResult</span><span class="op">$</span><span class="va">Description</span> <span class="op">&lt;-</span> <span class="fu">gsub</span><span class="op">(</span><span class="st">" - Mus musculus \\(house mouse)"</span>,<span class="st">""</span>,<span class="va">enrich_pathway</span><span class="op">@</span><span class="va">compareClusterResult</span><span class="op">$</span><span class="va">Description</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot top enriched functions using dotplot function of
clusterProfiler package.</p>
<div class="codewrapper sourceCode" id="cb95">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu">dotplot</span><span class="op">(</span><span class="va">enrich_pathway</span>,showCategory<span class="op">=</span><span class="fl">10</span>,font.size <span class="op">=</span> <span class="fl">14</span>,title<span class="op">=</span><span class="st">"Enriched KEGG Pathways"</span><span class="op">)</span></span></code></pre>
</div>
<p>What does this plot infer?</p>
</div>
<div class="section level2">
<h2 id="save-data-for-next-lesson">Save Data for Next Lesson<a class="anchor" aria-label="anchor" href="#save-data-for-next-lesson"></a>
</h2>
<p>We will use the results data in the next lesson. Save it now and we
will load it at the beginning of the next lesson. We will use R’s save
command to save the objects in compressed, binary format. The save
command is useful when you want to save multiple objects in one
file.</p>
<div class="codewrapper sourceCode" id="cb96">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">save</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>,<span class="va">DE_5xFAD.list</span>,file<span class="op">=</span><span class="st">"results/DEAnalysis_5XFAD.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="codewrapper sourceCode" id="cb97">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessionInfo</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
 [1] compiler_4.4.1  Matrix_1.6-5    tools_4.4.1     yaml_2.3.9
 [5] grid_4.4.1      knitr_1.48      xfun_0.46       renv_1.0.7
 [9] lattice_0.22-5  evaluate_0.24.0</code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</div></section><section id="aio-mouse-human-alignment"><p>Content from <a href="mouse-human-alignment.html">Mouse-human alignment of transcriptomic signatures</a></p>
<hr>
<p>Last updated on 2024-07-25 |

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/episodes/mouse-human-alignment.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 100 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do we perform a cross-species comparison?</li>
<li>Which aspects of disease does a model capture?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Approaches to align mouse data to human data</li>
<li>Understand the human AD co-expression modules</li>
<li>Perform correlation analysis</li>
<li>Understand the biological domains of Alzheimer’s disease</li>
<li>Use biological domain annotations to compare between species</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Author: Gregory Cary, Jackson Laboratory</p>
<section><h2 class="section-heading" id="aligning-human-and-mouse-phenotype">Aligning Human and Mouse Phenotype<a class="anchor" aria-label="anchor" href="#aligning-human-and-mouse-phenotype"></a>
<a class="anchor" aria-label="anchor" href="#aligning-human-and-mouse-phenotype"></a>
</h2>
<hr class="half-width">
<p>Alzheimer’s Disease (AD) is complex, and we can not expect any animal
model to fully recapitulate all aspects of late onset AD (LOAD)
pathology. To study AD with animal models we must find dimensions
through which we can align phenotypes between the models and human
cohorts. In MODEL-AD we use the following data modalities to identify
commonalities between mouse models and human cohorts:</p>
<ol style="list-style-type: decimal">
<li>Imaging (i.e. MRI and PET) to correspond with human imaging studies
(e.g.  <a href="https://adni.loni.usc.edu/" class="external-link">ADNI</a>)<br>
</li>
<li>Neuropathological phenotypes<br>
</li>
<li>
<em>Lots</em> of ’omics — genomics, proteomics, and
metabolomics</li>
</ol>
<p>The ’omics comparisons allow for a much richer contrast, since a
significant portion of genes are shared between these two species.
Homology at the anatomical and neuropathological levels is less
clear.</p>
<figure><img src="../fig/rmd04-dimensions_of_model_alignment.png" alt="Aligning Human and Mouse Phenotypes" class="figure mx-auto d-block"><div class="figcaption">Aligning Human and Mouse Phenotypes</div>
</figure></section><section><h2 class="section-heading" id="overview-of-human-cohort-data">Overview of Human cohort data<a class="anchor" aria-label="anchor" href="#overview-of-human-cohort-data"></a>
<a class="anchor" aria-label="anchor" href="#overview-of-human-cohort-data"></a>
</h2>
<hr class="half-width">
<p>The Accelerating Medicines Partnership-Alzheimer’s Disease Consortium
<a href="https://adknowledgeportal.synapse.org/Explore/Programs/DetailsPage?Program=AMP-AD" class="external-link">(AMP-AD)</a>
has generated extensive sets of ’omics data from a variety of human
Alzheimer’s disease cohorts. AMP-AD researchers are applying systems
biology approaches toward the goal of elucidating AD mechanisms and
highlighting potential therapeutic targets.</p>
<p>There are three large, independent human cohorts that are part of
AMP-AD:</p>
<ol style="list-style-type: decimal">
<li>The Religious Orders Study and the Memory and Aging Project (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3219045" class="external-link">ROSMAP,
syn3219045</a>)</li>
<li>Mount Sinai Brain Bank (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3159438" class="external-link">MSBB,
syn3159438</a>)<br>
</li>
<li>Mayo Clinic (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn5550404" class="external-link">Mayo,
syn5550404</a>)</li>
</ol>
<p>These studies have collected postmortem RNA-seq profiles from over
1,200 individuals spanning seven distinct brain regions: - dorsolateral
prefrontal cortex (DLPFC)<br>
- temporal cortex (TCX)<br>
- inferior frontal gyrus (IFG)<br>
- superior temporal gyrus (STG)<br>
- frontal pole (FP)<br>
- parahippocampal gyrus (PHG)<br>
- cerebellum (CBE)</p>
<p>These samples are generally balanced for AD, MCI, and non-affected
controls. The data provide a broad assessment on how AD affects multiple
brain regions in 3 different populations in the US.</p>
<figure><img src="../fig/rmd04-ampad_studies_regions.png" alt="Transcriptomic studies of Alzheimer’s Disease" class="figure mx-auto d-block"><div class="figcaption">Transcriptomic studies of Alzheimer’s
Disease</div>
</figure><p>Today we’ll consider two different approaches to compare gene
expression between human cohorts and model systems:</p>
<ol style="list-style-type: decimal">
<li>Correlation within co-expression module genes</li>
</ol>
<ul>
<li>correlations will be generally weak for all expression, but animal
models may recapitulate specific aspects of the disease</li>
<li>we can use subsets of genes from co-expression modules, which
represent genes expressed in similar patterns in AD, and look for
correlations within these subsets</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Correlation of functional enrichment results</li>
</ol>
<ul>
<li>another approach is to consider the functional annotation enriched
among differentially expressed genes in human and mouse</li>
<li>we can similarly sub-divide these groups of co-functional genes into
biological domains to aid our interpretation</li>
</ul></section><section><h2 class="section-heading" id="overview-of-human-consensus-rna-seq-coexpression-modules">Overview of Human Consensus RNA-Seq Coexpression Modules<a class="anchor" aria-label="anchor" href="#overview-of-human-consensus-rna-seq-coexpression-modules"></a>
<a class="anchor" aria-label="anchor" href="#overview-of-human-consensus-rna-seq-coexpression-modules"></a>
</h2>
<hr class="half-width">
<p><a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et
al.</a> performed meta analysis including all available AMP-AD RNA-seq
datasets and systematically defined correspondences between gene
expression changes associated with AD in human brains. Briefly, the
RNA-seq read libraries were normalized and covariates were adjusted for
each human study separately before testing for differential expression
using fixed/mixed effects modeling to account for batch effects. The
expression data from each brain region was used to perform co-expression
analysis using a variety of different algorithms, generating in total
2,978 co-expression modules across all tissues. Of these, 660 modules
showed enrichment for at least one AD-specific differentially expressed
gene from the meta-analysis of all cases compared to controls.</p>
<p>Wan <em>et al</em> clustered these modules together using network
analyses and found 30 co-expression modules related to LOAD pathology.
Among the 30 aggregate co-expression modules, five consensus clusters
were described that span brain region and study cohorts. These consensus
clusters consist of subsets of modules which are associated with similar
AD related changes across brain regions and cohorts. Wan <em>et al</em>
looked for enrichment of cell -type signatures within the modules using
expression-weighted cell type enrichment analysis (<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">Skene and Grant,
2016</a>) and examined enrichment of functional annotations within the
modules.</p>
<figure><img src="../fig/rmd04-consensus_clusters.png" alt="AMP-AD GENE Modules" class="figure mx-auto d-block"><div class="figcaption">AMP-AD GENE Modules</div>
</figure><p>This figure shows a matrix view of gene content overlap between the
30 co-expression modules. You’ll note a few strongly overlapping group
of modules, implicating similar expression profiles in different studies
and brain regions, which are the consensus clusters (A-E).</p>
<p>The first module block (consensus cluster A) is enriched for
signatures of astrocytes, while the next block (consensus cluster B) is
enriched for signatures of other cell types, including endothelial and
microglial expressed genes, suggesting an inflammation component. The
third module block (consensus cluster C) is strongly enriched for
signatures of neuronal gene expression, linking the modules within this
cluster to neurodegenerative processes. The fourth module block
(consensus cluster D) is enriched for oligodendrocyte and other glial
genes, indicating myelination and other neuronal support functions
associated with these modules. Finally, consensus cluster E contains
mixed modules that don’t have clear cell type enrichments, but do have
enrichments for processes involved in response to stress or unfolded
proteins. Stress response is not cell specific, so the expression
represented by these modules may be throughout many cells in the
brain.</p>
</section><section><h2 class="section-heading" id="accessing-amp-ad-module-data">Accessing AMP-AD module data<a class="anchor" aria-label="anchor" href="#accessing-amp-ad-module-data"></a>
<a class="anchor" aria-label="anchor" href="#accessing-amp-ad-module-data"></a>
</h2>
<hr class="half-width">
<p>These AMP-AD co-expression modules are very useful for making
comparisons between animal models and the human cohorts. In order to use
these modules to make the comparisons, we’ll need to download data
pertaining to the 30 co-expression modules. These data are available
from the <a href="https://www.synapse.org/Synapse:syn11932957" class="external-link">Synapse
data repository</a>; let’s download and take a closer look at the
data.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">synTableQuery</span><span class="op">(</span><span class="st">"SELECT * FROM syn11932957"</span><span class="op">)</span></span>
<span><span class="va">module_table</span> <span class="op">&lt;-</span> <span class="fu">read.table</span><span class="op">(</span><span class="va">query</span><span class="op">$</span><span class="va">filepath</span>, sep <span class="op">=</span> <span class="st">","</span>,header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at module table</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">module_table</span><span class="op">)</span></span></code></pre>
</div>
<p>Here you see 9 columns in this table. Columns we’re interested in
are:</p>
<ul>
<li>column 3: <code>GeneID</code> contains Ensembl gene IDs<br>
</li>
<li>column 4: <code>Module</code> is the module name in which gene is
clustered<br>
</li>
<li>column 7: <code>brainRegion</code> is the tissue of the
corresponding module<br>
</li>
<li>column 9: <code>external_gene_name</code> are gene symbols</li>
</ul>
<p>How many unique modules are in the table?</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>What are the names of the modules?</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
<p>How many genes are in each module?</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">table</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
<p>We can visualize this as bar plot using ggplot2 package.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">module_table</span>,<span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">Module</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> </span></code></pre>
</div>
<blockquote>
<h2 id="challenge-1">Challenge 1</h2>
<p>What are other ways to count genes in each module?</p>
<blockquote>
<h2 id="solution-to-challenge-1">Solution to Challenge 1</h2>
<pre><code><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">module_table</span> ,<span class="va">Module</span><span class="op">)</span></span></code></pre>
<p>{: .language-r} {: .solution} {: .challenge}</p>
</blockquote>
</blockquote>
<p>We can also check the total number of unique genes in the table</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">GeneID</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="mouse-orthologs-of-human-module-genes">Mouse orthologs of Human module genes<a class="anchor" aria-label="anchor" href="#mouse-orthologs-of-human-module-genes"></a>
</h4>
<p>In the module table we’ve downloaded we have human ENSEMBL ids and
human gene symbols. In order to compare between human and mouse models,
we will need to identify the corresponding mouse gene IDs. We are going
to add the gene IDs of orthologous genes in mouse to the corresponding
human genes in the module table.</p>
<p>Orthology mapping can be tricky, but thankfully Wan <em>et al</em>
have already identified mouse orthologs for each of the human genes
using the HGNC Comparison of Orthology Predictions (<a href="https://www.genenames.org/tools/hcop/" class="external-link">HCOP</a>) tool. While there
are a variety of different ways to get data about gene orthology, for
the sake of simplicity we are going to read that table from <a href="https://doi.org/10.7303/syn17010253.1" class="external-link">Synapse</a>.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouse.human.ortho</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn17010253"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s take a look at the first few rows of this orthology table:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">mouse.human.ortho</span><span class="op">)</span></span></code></pre>
</div>
<p>There are 15 columns with various gene identifiers for each species.
We’ll add mouse gene symbols from the ortholog table to the module table
by matching the human ENSEMBL IDs from both tables.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">module_table</span><span class="op">$</span><span class="va">Mouse_gene_name</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">mouse.human.ortho</span><span class="op">$</span><span class="va">mouse_symbol</span><span class="op">[</span></span>
<span>    <span class="fu">match</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">GeneID</span>,<span class="va">mouse.human.ortho</span><span class="op">$</span><span class="va">human_ensembl_gene</span><span class="op">)</span></span>
<span>    <span class="op">]</span></span></code></pre>
</div>
<p>Taking a look at the module table, we can see the new column of mouse
orthologs we just added.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">module_table</span><span class="op">)</span></span></code></pre>
</div>
<p>Some genes don’t have identified orthologs. Also there’s some
redundant information. Let’s only keep the columns of interest and rows
that contain a mouse ortholog mapping:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules</span> <span class="op">&lt;-</span> <span class="va">module_table</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">distinct</span><span class="op">(</span>tissue <span class="op">=</span> <span class="va">brainRegion</span>, module <span class="op">=</span> <span class="va">Module</span>, gene <span class="op">=</span> <span class="va">GeneID</span>, <span class="va">Mouse_gene_name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Mouse_gene_name</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span></code></pre>
</div>
<p>Take a look at this new data table:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules</span><span class="op">)</span></span></code></pre>
</div>
<blockquote>
<h2 id="challenge-2">Challenge 2</h2>
<p>How many human genes are we removing that don’t have identified
orthologs?</p>
<blockquote>
<h2 id="solution-to-challenge-2">Solution to Challenge 2</h2>
<pre><code><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">module_table</span>, <span class="fu">is.na</span><span class="op">(</span><span class="va">Mouse_gene_name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">external_gene_name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nrow</span><span class="op">(</span><span class="op">)</span></span></code></pre>
<p>{: .language-r} {: .solution} {: .challenge}</p>
</blockquote>
</blockquote>
</div>
</section><section><h2 class="section-heading" id="reading-differential-expression-result-of-human-data-from-meta-analysis">Reading differential expression result of human data from
meta-analysis<a class="anchor" aria-label="anchor" href="#reading-differential-expression-result-of-human-data-from-meta-analysis"></a>
<a class="anchor" aria-label="anchor" href="#reading-differential-expression-result-of-human-data-from-meta-analysis"></a>
</h2>
<hr class="half-width">
<p>Now we know the genes that are in each AMP-AD co-expression cluster,
along with the ID of the corresponding orthologous gene in mouse. We’ll
also need to know how these genes change expression in AD.</p>
<p>We’ll download the results from differential expression meta-analysis
of reprocessed AMP-AD RNA-seq data (all 7 brain regions). Log fold
change values for human transcripts can be obtained from <a href="https://www.synapse.org/#!Synapse:syn14237651" class="external-link">Synapse</a>.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_raw</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn14237651"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s take a look at these data</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_raw</span><span class="op">)</span></span></code></pre>
</div>
<p>Data from which tissues are in this table?</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="va">ampad_modules_raw</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span></span></code></pre>
</div>
<p>All 7 brain regions are represented.</p>
<p>The AMP-AD data has been processed many ways and using different
models and comparisons. Let’s take a look at how many ways the data have
been analyzed:</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Model</span>, <span class="va">Comparison</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>For our analyses we’ll use data from the “Diagnosis” model and
comparisons between AD cases vs controls. We’ll filter the table for
these conditions and only keep the three columns we’ll need:
<code>Tissue</code>, <code>Gene</code> and <code>logFC</code>.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_fc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span>tissue <span class="op">=</span> <span class="va">Tissue</span>, gene <span class="op">=</span> <span class="va">ensembl_gene_id</span>, ampad_fc <span class="op">=</span> <span class="va">logFC</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="combine-with-modules-so-correlation-can-be-done-per-module">Combine with modules so correlation can be done per module<a class="anchor" aria-label="anchor" href="#combine-with-modules-so-correlation-can-be-done-per-module"></a>
</h4>
<p>Next, we will combine the fold change table we just downloaded
(<code>ampad_fc</code>) and module table from before
(<code>ampad_modules</code>). First, let’s look at both tables to check
how can we merge them together?</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_fc</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules</span><span class="op">)</span></span></code></pre>
</div>
<p>The columns common to both tables are <code>gene</code> (the human
Ensembl gene IDs) and <code>tissue</code> (the brain region
corresponding to the module/measurement). So we will merge the data sets
using these two columns.</p>
<p><strong>Reminder</strong>: Each gene can be present in multiple brain
regions, but should only be in one module per brain region. Let’s double
check using the first gene in the table:</p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules</span><span class="op">[</span><span class="va">ampad_modules</span><span class="op">$</span><span class="va">gene</span> <span class="op">%in%</span> <span class="st">"ENSG00000168439"</span>,<span class="op">]</span></span></code></pre>
</div>
<p>This gene is present in six different co-expression modules all from
different brain regions. You can do this for any other gene as well.</p>
<p>We’ll merge the two tables using the <code>dplyr::inner_join</code>
function:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_fc</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_modules</span>, <span class="va">ampad_fc</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"tissue"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Mouse_gene_name</span>, <span class="va">module</span>, <span class="va">ampad_fc</span><span class="op">)</span> </span></code></pre>
</div>
<p>Take a look at the new table we just made:</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>We will use the data in <code>ampad_modules_fc</code> to compare with
log fold change data measured in mouse models.</p>
</div>
<div class="section level4">
<h4 id="preparing-module-information-for-correlation-plots">Preparing module information for correlation plots<a class="anchor" aria-label="anchor" href="#preparing-module-information-for-correlation-plots"></a>
</h4>
<p>Let’s package up these data and save this progress so far. This is
some manual book-keeping to arrange the modules into consensus clusters
for plotting later. You can just copy-paste this code into your
<code>Rstudio</code> session.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cluster_a</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"TCXblue"</span>, <span class="st">"PHGyellow"</span>, <span class="st">"IFGyellow"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster A (ECM organization)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster A\n(ECM organization)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_b</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"DLPFCblue"</span>, <span class="st">"CBEturquoise"</span>, <span class="st">"STGblue"</span>, <span class="st">"PHGturquoise"</span>, <span class="st">"IFGturquoise"</span>, <span class="st">"TCXturquoise"</span>, <span class="st">"FPturquoise"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster B (Immune system)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster B\n(Immune system)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_c</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"IFGbrown"</span>, <span class="st">"STGbrown"</span>, <span class="st">"DLPFCyellow"</span>, <span class="st">"TCXgreen"</span>, <span class="st">"FPyellow"</span>, <span class="st">"CBEyellow"</span>, <span class="st">"PHGbrown"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster C (Neuronal system)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster C\n(Neuronal system)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_d</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"DLPFCbrown"</span>, <span class="st">"STGyellow"</span>, <span class="st">"PHGgreen"</span>, <span class="st">"CBEbrown"</span>, <span class="st">"TCXyellow"</span>, <span class="st">"IFGblue"</span>, <span class="st">"FPblue"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster D (Cell Cycle, NMD)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster D\n(Cell Cycle, NMD)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_e</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"FPbrown"</span>, <span class="st">"CBEblue"</span>, <span class="st">"DLPFCturquoise"</span>, <span class="st">"TCXbrown"</span>, <span class="st">"STGturquoise"</span>, <span class="st">"PHGblue"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster E (Organelle Biogensis, Cellular stress response)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster E\n(Organelle Biogenesis,\nCellular stress response)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">module_clusters</span> <span class="op">&lt;-</span> <span class="va">cluster_a</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_b</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_c</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_d</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_e</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cluster_label <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">cluster_label</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="va">module_clusters</span><span class="op">$</span><span class="va">module</span></span>
<span></span>
<span><span class="fu">save</span><span class="op">(</span><span class="va">ampad_modules_fc</span>,<span class="va">module_clusters</span>,<span class="va">mod</span>,file<span class="op">=</span><span class="st">"../data/AMPAD_Module_Data.RData"</span><span class="op">)</span></span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-human-ad-modules">Correlation between mouse models and human AD modules<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-human-ad-modules"></a>
<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-human-ad-modules"></a>
</h2>
<hr class="half-width">
<p>As demonstrated in the plots below, we’ll aim to identify modules
where the gene expression is correlated between human and mouse
orthologs (left) as well as modules where there is no correlation
(right).</p>
<figure><img src="../fig/rmd04-hs_mm_logfc_correlation.png" alt="Human-mouse logFCcorrelation" class="figure mx-auto d-block"><div class="figcaption">Human-mouse logFCcorrelation</div>
</figure><p>There are two approaches that we commonly use to compute correlation
between mouse data and human AD data:</p>
<ol style="list-style-type: decimal">
<li>Compare change in expression in Human AD cases vs controls with
change in expression in mouse models for each gene in a given
module:<br>
</li>
</ol>
<ul>
<li>LogFC(h) = log fold change in expression of human AD patients
compared to control patients.</li>
<li>LogFC(m) = log fold change in expression of mouse AD models compared
to control mouse models.</li>
</ul>
<p><span class="math display">\[cor.test(LogFC(h),
LogFC(m))\]</span></p>
<ol start="2" style="list-style-type: decimal">
<li>Compare Human AD expression changes to mouse genetic effects for
each gene in a given module:<br>
</li>
</ol>
<ul>
<li>h = human gene expression (Log2 RNA-seq Fold Change AD/control)</li>
<li>β = mouse gene expression effect from linear regression model (Log2
RNA-seq TPM)</li>
</ul>
<p><span class="math display">\[cor.test(LogFC(h), β)\]</span></p>
<p>Both approaches allow us to assess directional coherence between gene
expression for genes in AMP-AD modules and the effects of genetic
manipulations in mice. For this lesson we are going to use the first
approach.</p>
<div class="section level4">
<h4 id="step-0-reading-gene-expression-count-matrix-from-previous-lesson">Step 0: Reading Gene Expression Count matrix from Previous
Lesson<a class="anchor" aria-label="anchor" href="#step-0-reading-gene-expression-count-matrix-from-previous-lesson"></a>
</h4>
<p>We’ll first read the result table saved after differential analysis
in last <a href="https://asliuyar.github.io/omicsTranslationAD/03-DESeq/index.html" class="external-link">lesson</a>.
We’ll start with the data from the 5xFAD mouse model to understand the
required steps to perform the correlation with human AD modules.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"results/DEAnalysis_5XFAD.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'results/DEAnalysis_5XFAD.Rdata', probable reason 'No such file or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<p>We can also load AMP-AD module data.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"data/AMPAD_Module_Data.RData"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-1-measure-the-correlation-between-mouse-models-for-each-sex-at-each-age-and-amp-ad-modules-using-common-genes-from-both-datasets">Step 1: Measure the correlation between mouse models for each sex at
each age and AMP-AD modules using common genes from both datasets<a class="anchor" aria-label="anchor" href="#step-1-measure-the-correlation-between-mouse-models-for-each-sex-at-each-age-and-amp-ad-modules-using-common-genes-from-both-datasets"></a>
</h4>
<p>We compute a Pearson correlation between changes in expression for
each gene within a given module (log fold change for cases vs controls)
with each mouse model (log fold change of the 5xFAD mice vs sex- and
age-matched B6 mice).</p>
<p>First, we’ll combine both mouse <code>DE_5xFAD.df</code> and human
<code>ampad_modules_fc</code> log fold change data sets for all
genes.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_ampad</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>, <span class="va">ampad_modules_fc</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span>, multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> </span></code></pre>
</div>
<p><strong>Note:</strong> for this join we specify
<code>multiple = "all"</code> so that the same gene can be matched
across multiple human tissues.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">model_vs_ampad</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we’ll create a list column containing data frames using <a href="https://tidyr.tidyverse.org/reference/nest.html" class="external-link">tidyr::nest</a>
function. Nesting is implicitly a summarizing operation: you get one row
for each group defined by the non-nested columns.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">model_vs_ampad</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">module</span>, <span class="va">model</span>, <span class="va">sex</span>, <span class="va">age</span>, <span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">ampad_fc</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">module</span>, <span class="va">model</span>, <span class="va">sex</span>, <span class="va">age</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s take a look:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>And we can also look at some of the nested data:</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>These are the mouse and human log fold-change values for all genes in
the TCXblue module; the mouse data corresponds to 4 month old male 5xFAD
mice.</p>
<p>The total number of groups in the data table:</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we’ll compute correlation coefficients using the
<code>cor.test</code> function:</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">df</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>      cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"log2FoldChange"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                      <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>      p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<p>Here we’re using <code>purrr::map</code> based functions to apply the
correlation test to every entry in the <code>data</code> column. We can
pull out specific features from the <code>cor_test</code> list column,
including the computed correlation coefficient (<code>estimate</code>)
and the significance (<code>p.value</code>).</p>
<p>In the end we should have correlation coefficients and significance
values for every comparison in our data table:</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-2-annotate-correlation-table-to-prepare-for-visualization">Step 2: Annotate correlation table to prepare for visualization<a class="anchor" aria-label="anchor" href="#step-2-annotate-correlation-table-to-prepare-for-visualization"></a>
</h4>
<p>These steps will make it easier to make a nice looking plot during
the next step. We’ll add a column that flags whether the correlation is
significant or not, and we’ll add in the information about which
consensus cluster each module belongs to:</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_module</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_clusters</span>, by <span class="op">=</span> <span class="st">"module"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">module</span>, <span class="va">model</span>, <span class="va">sex</span>, <span class="va">age</span>,</span>
<span>                correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">model_module</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-3-create-a-dataframe-to-use-as-input-for-plotting-the-results">Step 3: Create a dataframe to use as input for plotting the
results<a class="anchor" aria-label="anchor" href="#step-3-create-a-dataframe-to-use-as-input-for-plotting-the-results"></a>
</h4>
<p>More preparations for plotting, here we’ll get all of the values in
the right order so that they are grouped together nicely on the
plot.</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">order.model</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5xFAD (Male)"</span>, <span class="st">"5xFAD (Female)"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_for_plot</span> <span class="op">&lt;-</span> <span class="va">model_module</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>      module <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">module</span>,levels<span class="op">=</span><span class="va">mod</span><span class="op">)</span>,</span>
<span>      model_sex <span class="op">=</span> <span class="fu">glue</span><span class="fu">::</span><span class="fu">glue</span><span class="op">(</span><span class="st">"{model} ({str_to_title(sex)})"</span><span class="op">)</span>,</span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">model_sex</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>      age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4 mo'</span>,<span class="st">'6 mo'</span>, <span class="st">'12 mo'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      model_sex <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">model_sex</span>,levels<span class="op">=</span><span class="va">order.model</span><span class="op">)</span>,</span>
<span>      model_sex <span class="op">=</span> <span class="fu">fct_rev</span><span class="op">(</span><span class="va">model_sex</span><span class="op">)</span>,</span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">correlation_for_plot</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="visualizing-the-correlation-plot">Visualizing the Correlation plot<a class="anchor" aria-label="anchor" href="#visualizing-the-correlation-plot"></a>
</h3>
<p>Now, we will use the above matrix and visualize the correlation
results using <code>ggplot2</code> package functions.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">correlation_for_plot</span></span>
<span><span class="fu">range</span><span class="op">(</span><span class="va">correlation_for_plot</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># the AMP-AD modules will be along the x-axis, the mouse models will be along the y-axis</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">model_sex</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># each tile of the grid will be filled with a circle where the fill and size correspond to the correlation coefficient</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">model_sex</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># we'll draw a box arround significant correlations</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">model_sex</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># plot the x-axis on the top of the plot, set the parameters of the scales</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># remove axis labels</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># facet the plot based on age range for the mice (rows) and consensus cluster (columns)</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">facet_grid</span><span class="op">(</span> rows <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">age</span><span class="op">)</span>,cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># specify how different aspects of the plot will look</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">theme</span><span class="op">(</span></span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<figure><img src="../results/AMPAD1.png" alt="AMPAD - 5XFAD correlation plot" class="figure mx-auto d-block"><div class="figcaption">AMPAD - 5XFAD correlation plot</div>
</figure><p>In the above plot, categories along the x-axis are the 30 AMP-AD
co-expression modules grouped into 5 consensus clusters, while the
categories along the y-axis show the different groupings of mouse models
tested (split by age and sex). Positive correlations are shown in blue
and negative correlations in red. Color intensity and size of the
circles are proportional to the correlation coefficient. Black squares
around dots represent significant correlation at p-value=0.05 and
non-significant correlations are left blank.</p>
<p>Male and female 5xFAD mice display gene expression alterations across
all five consensus clusters, with the strongest correlations observed
among modules/genes in Consensus Cluster B, which generally consists of
immune system pathways and functions.</p>
</div>
<div class="section level3">
<h3 id="examining-individual-correlation-results">Examining individual correlation results<a class="anchor" aria-label="anchor" href="#examining-individual-correlation-results"></a>
</h3>
<p>Let’s say we want to know more about a single comparison in the plot
above and which genes are contributing to the correlation result. Maybe
we’re really interested in the correlations to the FPbrown module
between female 5xFAD mice at 4 months and 6 months. We can plot the
individual correlations for each comparison in the plot above with the
data we have. We’ll label genes with large fold change in the mouse
using the <code>ggrepel::geom_label_repel</code> function.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specify which comparisons to consider</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="st">'FPbrown'</span></span>
<span><span class="va">s</span> <span class="op">&lt;-</span> <span class="st">'female'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4 mo'</span>,<span class="st">'6 mo'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the correlation data frame to these comparisons</span></span>
<span><span class="va">indiv.corr</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">m</span>, <span class="va">sex</span> <span class="op">==</span> <span class="va">s</span>, <span class="va">age</span> <span class="op">%in%</span> <span class="va">a</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> facet <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'\n'</span>, <span class="st">'r = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">estimate</span>,<span class="fl">3</span><span class="op">)</span>,<span class="st">' ; p = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">p_value</span>,<span class="fl">3</span><span class="op">)</span> <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">aes</span><span class="op">(</span> <span class="va">log2FoldChange</span> , <span class="va">ampad_fc</span> <span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span> size <span class="op">=</span> <span class="fl">.5</span>, color <span class="op">=</span> <span class="st">'darkred'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, lwd <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_label_repel</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">desc</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                             <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'5xFAD logFC'</span>, y <span class="op">=</span> <span class="st">'AMP-AD logFC'</span>,</span>
<span>       title <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">indiv.corr</span><span class="op">$</span><span class="va">module</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">facet</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../results/AMPAD_FPbrown.png" alt="AMPAD - 5XFAD FPbrown correlation" class="figure mx-auto d-block"><div class="figcaption">AMPAD - 5XFAD FPbrown correlation</div>
</figure><p>Here we can see that the 5xFAD logFC are pretty small in general. The
correlations are relatively weak and driven by individual genes that
have relatively large changes (e.g. Heatr4, Nirp1a, Rpl39l). If we
compare a different module, say STGblue, we can see a stronger
relationship between the mouse and human expression.</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="st">'STGblue'</span></span>
<span><span class="va">s</span> <span class="op">&lt;-</span> <span class="st">'female'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4 mo'</span>,<span class="st">'12 mo'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">indiv.corr</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">m</span>, <span class="va">sex</span> <span class="op">==</span> <span class="va">s</span>, <span class="va">age</span> <span class="op">%in%</span> <span class="va">a</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> v <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'\n'</span>, <span class="st">'r = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">estimate</span>,<span class="fl">3</span><span class="op">)</span>,<span class="st">' ; p = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">p_value</span>,<span class="fl">3</span><span class="op">)</span> <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">aes</span><span class="op">(</span> <span class="va">log2FoldChange</span> , <span class="va">ampad_fc</span> <span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span> size <span class="op">=</span> <span class="fl">.5</span>, color <span class="op">=</span> <span class="st">'darkred'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, lwd <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_label_repel</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">desc</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                             <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'5xFAD logFC'</span>, y <span class="op">=</span> <span class="st">'AMP-AD logFC'</span>,</span>
<span>       title <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">indiv.corr</span><span class="op">$</span><span class="va">module</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">v</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../results/AMPAD_STGblue.png" alt="AMPAD - 5XFAD STGblue correlation" class="figure mx-auto d-block"><div class="figcaption">AMPAD - 5XFAD STGblue correlation</div>
</figure><p>These correlations are much stronger (r ~ 0.3 vs 0.1 in the previous
example), and there is a consistent pattern between young mice and old
mice, with similar genes being expressed in similar ways (e.g. Itgax and
Clec7a are up-regulated at both ages). We also saw in the Volcano plot
that immune-related genes were significantly up-regulated in the 5xFAD
model. These significant positive correlation suggests that the 5xFAD
model captures inflammatory changes observed in human AD patients.</p>
<blockquote>
<h2 id="challenge-3">Challenge 3</h2>
<p>Considering the LOAD1 (i.e. APOE4.Trem2) model results, which modules
show correlation and how does it compare with 5xFAD?</p>
<blockquote>
<h2 id="solution-to-challenge-2-1">Solution to Challenge 2</h2>
<pre><code>load("../results/DEAnalysis_LOAD1.Rdata")
model_vs_ampad &lt;- inner_join(DE_LOAD1.df, ampad_modules_fc, by = c("symbol"), multiple = "all")
cor.df &lt;- model_vs_ampad %&gt;%
  dplyr::select(module, model, sex, age, symbol, log2FoldChange, ampad_fc) %&gt;%
  group_by(module, model, sex, age) %&gt;%
  nest(data = c(symbol, log2FoldChange, ampad_fc)) %&gt;%
  mutate(
      cor_test = map(data, ~ cor.test(.x[["log2FoldChange"]],
                                      .x[["ampad_fc"]], method = "pearson")),
      estimate = map_dbl(cor_test, "estimate"),
      p_value = map_dbl(cor_test, "p.value")
      ) %&gt;%
  ungroup() %&gt;%
  dplyr::select(-cor_test)
model_module &lt;- cor.df %&gt;%
    mutate(significant = p_value &lt; 0.05) %&gt;%
    left_join(module_clusters, by = "module") %&gt;%
    dplyr::select(cluster, cluster_label, module, model, sex, age,
                  correlation = estimate, p_value, significant)
order.model &lt;- c("APOE4 (Male)", "APOE4 (Female)","Trem2 (Male)", "Trem2 (Female)","APOE4Trem2 (Male)", "APOE4Trem2 (Female)")
correlation_for_plot &lt;- model_module %&gt;%
    arrange(cluster) %&gt;%
    mutate(
      module = factor(module,levels=mod),
      model_sex = glue::glue("{model} ({str_to_title(sex)})"),
   ) %&gt;%
    arrange(model_sex) %&gt;%
    mutate(
      age = factor(age, levels = c(4,8)),
      model_sex = factor(model_sex,levels=order.model),
      model_sex = fct_rev(model_sex),
    )
data &lt;- correlation_for_plot
ggplot2::ggplot() +
  ggplot2::geom_tile(data = data, ggplot2::aes(x = .data$module, y = .data$model_sex), colour = "black", fill = "white") +
  ggplot2::geom_point(data = dplyr::filter(data), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = &gt; &gt; .data$correlation, size = abs(.data$correlation))) +
  ggplot2::geom_point(data = dplyr::filter(data, .data$significant), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation),color="black",shape=0,size=9) +
  ggplot2::scale_x_discrete(position = "top") +
  ggplot2::scale_size(guide = "none", limits = c(0, 0.4)) +
  ggplot2::scale_color_gradient2(limits = c(-0.5, 0.5), low = "#85070C", high = "#164B6E", name = "Correlation", guide = ggplot2::guide_colorbar(ticks = FALSE)) +
  ggplot2::labs(x = NULL, y = NULL) +
  ggplot2::facet_grid( rows = dplyr::vars(.data$age),cols = dplyr::vars(.data$cluster_label), scales = "free", space = "free",switch = 'y') +
  ggplot2::theme(
    strip.text.x = ggplot2::element_text(size = 10,colour = c("black")),
    strip.text.y.left = ggplot2::element_text(angle = 0,size = 12),
    axis.ticks = ggplot2::element_blank(),
    axis.text.x = ggplot2::element_text(angle = 90, hjust = 0, size = 12),
    axis.text.y = ggplot2::element_text(angle = 0, size = 12),
    panel.background = ggplot2::element_blank(),
    plot.title = ggplot2::element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
    plot.title.position = "plot",
    panel.grid = ggplot2::element_blank(),
    legend.position = "right"
  )</code></pre>
<p>{: .language-r} The LOAD1 models show weaker correlation to AMP-AD
module gene expression overall, and in particular anti-correlation for
Consensus Cluster B (immune) modules. {: .solution} {: .challenge}</p>
</blockquote>
</blockquote>
</div>
</section><section><h2 class="section-heading" id="detecting-functional-coherence-of-gene-sets-from-omics-data">Detecting functional coherence of gene sets from omics data<a class="anchor" aria-label="anchor" href="#detecting-functional-coherence-of-gene-sets-from-omics-data"></a>
<a class="anchor" aria-label="anchor" href="#detecting-functional-coherence-of-gene-sets-from-omics-data"></a>
</h2>
<hr class="half-width">
<p>Most omics analyses produce data on many thousands of genomic
features (i.e.  transcripts, proteins, etc.) for each condition tested.
Simply looking at these lists of genes and associated statistics can be
daunting and uninformative. We need approaches to identify which
biological functions are being impacted by a given experiment from these
systems-level measurements.</p>
<p>Gene functional enrichment analysis describes a variety of
statistical methods that identify groups of genes that share a
particular biological function or process and show differential
association with experimental conditions. Most approaches compare some
statistically valid set of differentially expressed features to sets of
functional annotations for those features. There are many different
functional annotation sets available, some of the more commonly used
include:</p>
<ul>
<li>gene function resources, such as the <a href="../amigo.geneontology.org/amigo">Gene Ontology</a> (i.e. GO)<br>
</li>
<li>pathway databases, such as <a href="../reactome.org">Reactome</a> or <a href="../www.genome.jp/kegg">KEGG</a><br>
</li>
<li>disease and phenotype ontologies, such as the <a href="../hpo.jax.org/app">Human Phenotype Ontology</a>, the <a href="../www.informatics.jax.org/vocab/mp_ontology">Mammalian Phenotype
Ontology</a>, and the <a href="https://disease-ontology.org/" class="external-link">Disease
Ontology</a>
</li>
</ul>
<p>These are the resources that are the foundation for many genomics
knowledge bases, such as <a href="../www.informatics.jax.org">MGI</a>, <a href="../monarchinitiative.org">monarch initiative</a>, and the <a href="../alliancegenome.org">Alliance of Genome Resources</a>. The precise
nature of each of these resources varies, but the core information
contained within each is the relationship of sets of genes to
biologically meaningful annotations. These annotations are typically
expertly curated from the published literature.</p>
<p>There are a variety of statistical approaches that can be employed to
test for functional enrichment among genes identified from an omics
dataset. Two of the most common broad classes of tests are
<em>over-representation analysis (ORA)</em> and <em>gene set enrichment
analysis (GSEA)</em>. For example, consider the figure below from <a href="https://doi.org/10.1016/j.tig.2023.01.003" class="external-link">Zhao &amp; Rhee, Trends
in Genetics (2023)</a>. Let’s consider each in a bit more detail.</p>
<figure><img src="../fig/rmd04-Zhao_Rhee_TrendsGenetics_pathway_enrichment.jpg" alt="Functional enrichment approaches" class="figure mx-auto d-block"><div class="figcaption">Functional enrichment approaches</div>
</figure><div class="section level3">
<h3 id="over-representation-analysis">Over-representation analysis<a class="anchor" aria-label="anchor" href="#over-representation-analysis"></a>
</h3>
<p><code>ORA</code> involves statistical tests of overlap between two
lists of genes: one derived from the experiment and one from the
functional annotations. For example, one might ask what is the overlap
between the genes in an annotation class, such as “Lysosome”, and the
genes that are significantly up-regulated in a given experimental
condition. These tests usually rely on some form of Fisher’s exact test
(e.g. <code>fisher.test()</code>) or hypergeometric test (e.g. 
<code>phyper()</code>). If the gene lists consist of a larger number of
overlapping genes than would be expected at random - given the sample
sizes involved - then there is said to be a statistical
over-representation of the annotation class within the experimental
condition.</p>
<p>Of course, these overlap tests are generally considered for all
annotation classes, which can number in the hundreds to thousands.
Performing this many statistical tests ensures that many will be
identified as significant by chance. Therefore, there is typically a
requirement to correct for multiple testing errors
(e.g. <code>p.adjust()</code>).</p>
<p>There are many R packages available to handle the statistical tests
and corrections involved in ORA. Today we’ll use
<code>clusterProfiler::enrichGO()</code>, which wraps statistical
testing for overlap with GO terms and multiple test correction in one
function.</p>
<p>Let’s start by considering the enrichments from the mouse data
analyzed previously.</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">theme_set</span><span class="op">(</span><span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in theme_set(theme_bw()): could not find function "theme_set"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">'results/DEAnalysis_5XFAD.RData'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'results/DEAnalysis_5XFAD.RData', probable reason 'No such file or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in readChar(con, 5L, useBytes = TRUE): cannot open the connection</code></pre>
</div>
<p>We’ll start by considering the genes that are significantly DE in 12
month old male animals</p>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list.up</span> <span class="op">&lt;-</span> <span class="va">DE_5xFAD.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">sex</span> <span class="op">==</span> <span class="st">'male'</span>, </span>
<span>         <span class="va">age</span> <span class="op">==</span> <span class="st">'12 mo'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, </span>
<span>         <span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene.list.dn</span> <span class="op">&lt;-</span> <span class="va">DE_5xFAD.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">sex</span> <span class="op">==</span> <span class="st">'male'</span>, </span>
<span>         <span class="va">age</span> <span class="op">==</span> <span class="st">'12 mo'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, </span>
<span>         <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="va">gene.list.up</span><span class="op">)</span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="va">gene.list.dn</span><span class="op">)</span></span></code></pre>
</div>
<p>There are a total of significantly up-regulated genes and
significantly down-regulated genes in this cohort. Now test for over
representation of GO terms among the DEGs. First, we need to identify
the universe of all possible genes which includes the genes that were
both measured by the experiment and contained within the annotation
set.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">univ</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">org.Mm.egGO</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bitr</span><span class="op">(</span><span class="va">.</span>, fromType <span class="op">=</span> <span class="st">"ENTREZID"</span>, toType <span class="op">=</span> <span class="st">'SYMBOL'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">intersect</span><span class="op">(</span><span class="va">.</span>, <span class="va">DE_5xFAD.df</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<p>Now let’s test for enriched GO terms (this can take 3-4 minutes)</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.up</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.up</span>, </span>
<span>                   ont <span class="op">=</span> <span class="st">'all'</span>, </span>
<span>                   OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>                   keyType <span class="op">=</span> <span class="st">'SYMBOL'</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">univ</span></span>
<span>                   <span class="op">)</span></span>
<span><span class="va">enr.dn</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.dn</span>, </span>
<span>                   ont <span class="op">=</span> <span class="st">'all'</span>, </span>
<span>                   OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>                   keyType <span class="op">=</span> <span class="st">'SYMBOL'</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">univ</span></span>
<span>                   <span class="op">)</span></span></code></pre>
</div>
<p>How many significant terms are identified:</p>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.up</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enr.dn</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<blockquote>
<h2 id="challenge-4">Challenge 4</h2>
<ol style="list-style-type: decimal">
<li>How many significant terms are identified from the up-regulated gene
list if you do not specify the “universe”?</li>
</ol>
<blockquote>
<h2 id="solution-to-challenge-1-1">Solution to Challenge 1</h2>
<pre><code><span><span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.up</span>, ont <span class="op">=</span> <span class="st">'all'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, keyType <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">.</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>  </span></code></pre>
<p>{: .language-r} {: .solution} {: .challenge}</p>
</blockquote>
</blockquote>
<p>Plot the top 10 significant terms:</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu">plot_grid</span><span class="op">(</span></span>
<span>  <span class="fu">dotplot</span><span class="op">(</span><span class="va">enr.dn</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'down'</span><span class="op">)</span>,</span>
<span>  <span class="fu">dotplot</span><span class="op">(</span><span class="va">enr.up</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'up'</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../results/significantTerms.png" alt="significantTerms" class="figure mx-auto d-block"><div class="figcaption">significantTerms</div>
</figure><p>From this we can see that nervous system related terms
(e.g. “learning or memory”, “neurotransmitter transport”) are down in
5xFAD males at 12 months, while immune functions
(e.g. “cytokine-mediated signaling pathway” and “leukocyte mediated
immunity”) are up in 5xFAD males at 12 months.</p>
</div>
<div class="section level3">
<h3 id="gene-set-enrichment-analysis">Gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis"></a>
</h3>
<p><code>GSEA</code> is an alternative approach that uses a statistical
measure from the omics data (e.g. the log fold change or significance)
to rank the genes. An “enrichment score” is calculated for each
annotation set based on where the genes annotated to the term sit in the
overall distribution.</p>
<p>Let’s analyze those same data with the
<code>clusterProfiler::gseGO()</code> function. First we’ll specify the
gene list and use the <code>log2FoldChange</code> value to rank the
genes in the list.</p>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list</span> <span class="op">&lt;-</span> <span class="va">DE_5xFAD.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">sex</span> <span class="op">==</span> <span class="st">'male'</span>, </span>
<span>         <span class="va">age</span> <span class="op">==</span> <span class="st">'12 mo'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">log2FoldChange</span>, name <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we’ll test for enrichment:</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr</span> <span class="op">&lt;-</span> <span class="fu">gseGO</span><span class="op">(</span><span class="va">gene.list</span>, ont <span class="op">=</span> <span class="st">'all'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, keyType <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span></span></code></pre>
</div>
<p>How many significant terms are identified:</p>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<blockquote>
<h2 id="challenge-5">Challenge 5</h2>
<ol style="list-style-type: decimal">
<li>The tally above represents all genes, both up- and down-regulatd. To
compare between GSEA and ORA, can you identify how many GSEA enriched
terms are associated with up-regulated genes and how many are associated
with down-regulated genes? (Hint: the <code>NES</code> value within the
results relates to the directionality of enrichment).</li>
</ol>
<blockquote>
<h2 id="solution-to-challenge-2-2">Solution to Challenge 2</h2>
<pre><code><span><span class="va">enr</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enr</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
<p>{: .language-r} {: .solution} {: .challenge}</p>
</blockquote>
</blockquote>
<p>Plot the term enrichments</p>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ridgeplot</span><span class="op">(</span><span class="va">enr</span>, core_enrichment <span class="op">=</span> <span class="cn">T</span>, showCategory <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../results/ridgePlot.png" alt="ridgeplot" class="figure mx-auto d-block"><div class="figcaption">ridgeplot</div>
</figure><p>This plot shows the top 15 GSEA enriched terms. The displayed terms
are all from the up-regulated class (i.e. <code>NES &gt; 0</code>), and
consist of many immune-relevant terms (e.g. “immune response” and
“cytokine production”).</p>
</div>
<div class="section level3">
<h3 id="common-pitfalls-best-practices">Common pitfalls &amp; best practices<a class="anchor" aria-label="anchor" href="#common-pitfalls-best-practices"></a>
</h3>
<p>These kinds of functional enrichment analyses are very common, but
not all results reported are equal! A recent paper describes an <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009935" class="external-link">“Urgent
need for consistent standards in functional enrichment analysis”</a>.
They examine how the results of functional enrichment analyses are
reported in the literature, and identify several common shortcomings.
Watch out for these common mistakes when performing and reporting your
own analyses!</p>
<figure><img src="../fig/rmd04-Wijesooriya_PLOS_CompBio_F2_cropped.png" alt="Common pitfalls" class="figure mx-auto d-block"><div class="figcaption">Common pitfalls</div>
</figure>
</div>
</section><section><h2 class="section-heading" id="alzheimerss-disease-biological-domains">Alzheimers’s Disease Biological Domains<a class="anchor" aria-label="anchor" href="#alzheimerss-disease-biological-domains"></a>
<a class="anchor" aria-label="anchor" href="#alzheimerss-disease-biological-domains"></a>
</h2>
<hr class="half-width">
<p>While these results are informative and help to provide essential
biological context to the results of the omics experiment, it is
difficult to understand all of the enriched terms and what that means
for the biology of disease. It would be useful to group each of the
enriched terms within broader areas of disease biology. There are
several common molecular endophenotypes that are detected within human
Alzheimer’s cohorts across omics data types (transcriptomics,
proteomics, and genetics). Several of these endophenotypic areas are
shown in the figure below (source: Jesse Wiley).</p>
<figure><img src="../fig/rmd04-AD_endophenotypes.png" alt="Endophenotypes of AD" class="figure mx-auto d-block"><div class="figcaption">Endophenotypes of AD</div>
</figure><p>These endophenotypes describe molecular processes that are
dysregulated by or during the disease process. Very similar sets of
endophenotypes have been cataloged among the targets of therapeutics in
clinical trials for AD (see below, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10210334/" class="external-link">Cummings
et al, Alzheimer’s disease drug development pipeline: 2023, Alz Dem
TRCI</a>.</p>
<figure><img src="../fig/rmd04-CADRO_2023.jpg" alt="Common AD Research Ontology" class="figure mx-auto d-block"><div class="figcaption">Common AD Research Ontology</div>
</figure><figure><img src="../fig/rmd04-CADRO_2023_2.jpg" alt="Drugs in Pipeline by CADRO" class="figure mx-auto d-block"><div class="figcaption">Drugs in Pipeline by CADRO</div>
</figure><p>In order to formalize and operationalize these endophenotypic areas,
the <a href="../treatad.org/emory-sage-sgc">Emory-Sage-SGC center of the
TREAT-AD consortium</a> has developed the <a href="https://alz-journals.onlinelibrary.wiley.com/doi/pdfdirect/10.1002/trc2.12461" class="external-link">Biological
Domains of AD</a>. In the enumeration of these domains, several criteria
were established; the biological domains defined should be:</p>
<ul>
<li>
<em>objective:</em> leverage existing well-annotated resources<br>
</li>
<li>
<em>automatable:</em> in CADRO, therapeutics and targets are
manually assigned<br>
</li>
<li>
<em>intelligible:</em> groupings should be easy to understand<br>
</li>
<li>
<em>modifiable:</em> definitions should be (and are!) continuously
be updated.</li>
</ul>
<p>In all, 19 distinct biological domains (aka biodomains) have been
identified. These biological domains are defined using sets of GO terms
that align with the endophenotype. For example, terms like “cytokine
receptor binding” and “leukocyte migration” are annotated to the
<code>Immune Response</code> biodomain, while terms like “postsynaptic
density” and “synaptic vesicle cycle” are annotated to the
<code>Synapse</code> biodomain. Of all terms in the GO, 7,127 terms
(16.4%) are annotated to one of the biological domains. Because the GO
terms have genes annotated, genes can be associated with specific
endophenotypes via the biological domain groupings of terms. In all,
18,289 genes are annotated to at least 1 biodomain term. While the
biodomains exhibit very few overlapping GO terms (panel A), due to gene
pleiotropy (etc) the number of overlapping genes between biological
domains is quite a bit higher (panel B). The development of the
biological domains is described in more detail in the published paper <a href="../doi.org/10.1002/trc2.12461">Genetic and Multi-omic Risk Assessment
of Alzheimer’s Disease Implicates Core Associated Biological
Domains</a>.</p>
<figure><img src="../fig/rmd04-score_domains_preprint_F1_cropped.jpeg" alt="AD Biodomain Demographics" class="figure mx-auto d-block"><div class="figcaption">AD Biodomain Demographics</div>
</figure><div class="section level3">
<h3 id="download-and-explore-the-biological-domain-annotations">Download and explore the biological domain annotations<a class="anchor" aria-label="anchor" href="#download-and-explore-the-biological-domain-annotations"></a>
</h3>
<p>First let’s download the biodomain definition file from synapse.</p>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#synLogin()</span></span>
<span></span>
<span><span class="co"># biodomain definitions</span></span>
<span><span class="va">biodom</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn25428992'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span><span class="va">dom.lab</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn26856828'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span></code></pre>
</div>
<p>What is in the file?</p>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">biodom</span><span class="op">)</span></span></code></pre>
</div>
<p>You can see the file is a list of GO term accessions
(<code>GO_ID</code>) and names (<code>GOterm_Name</code>) as well as the
corresponding endophenotype (<code>Biodomain</code>) to which the term
is annotated. There are also gene annotations for each term. These
annotation sets come from two sources: (1) the <code>symbol</code> and
<code>uniprot</code> annotations are drawn directly from the Gene
Ontology via the provided API, (2) <code>ensembl_id</code>,
<code>entrez_id</code>, and <code>hgnc_symbol</code> are from BioMart
annotations (e.g. <code>biomaRt::getLDS()</code>).</p>
<p>We can see how many GO terms are annotated to each biodomain:</p>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n_term <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">n_term</span><span class="op">)</span> </span></code></pre>
</div>
<p>The biodomains range in size from <code>Tau Homeostasis</code> (10
terms) up to <code>Synapse</code> (1,379 terms).</p>
<p>We can also investigate the individual genes annotated to each
biodomain GO term.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOterm_Name</span> <span class="op">==</span> <span class="st">'amyloid-beta formation'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unlist</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>So the genes associated with <code>amyloid-beta formation</code>
within the <code>APP Metabolism</code> biodomain include PSEN1, PSEN2,
BACE1, ABCA7, NCSTN, and others.</p>
</div>
<div class="section level3">
<h3 id="annotate-enriched-terms-with-biological-domain">Annotate enriched terms with biological domain<a class="anchor" aria-label="anchor" href="#annotate-enriched-terms-with-biological-domain"></a>
</h3>
<p>Let’s re-characterize the 5xFAD functional enrichments we computed
previously with the biological domain annotations and see if we can get
more context about what is changing in that model. We’ll focus on the
GSEA results and start by annotating the results with biodomain
groupings.</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd</span> <span class="op">=</span> <span class="va">enr</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, ID <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'ID'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">enr.bd</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>Not all of the enriched terms are annotated to a biological domain.
Some terms are too broad and not specific (e.g. ‘defense response’),
while others may not have been captured by a biological domain
annotation yet (e.g. ‘regulation of immune system process’). Remember
that the conception of the biodomains involved a requirement that they
be modifiable, and these terms may be added to the biodomain
definintions in the future. Let’s modify the <code>Biodomain</code>
value for terms that aren’t annotated to a domain to ‘none’.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">[</span><span class="fu">is.na</span><span class="op">(</span><span class="va">enr.bd</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'none'</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">enr.bd</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>How many terms are enriched from each domain?</p>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span> <span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">enr.bd</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span> <span class="va">enr.bd</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">n_sig_term</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Many enriched terms don’t map to a domain (), but most do (). Of
those that do, the vast majority map into the
<code>Immune Response</code> biodomain.</p>
<p>We can plot the enrichment results, stratified by biodomain:</p>
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="va">n_sig_term</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">p.adjust</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.adjust</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../results/biodom_plot.png" alt="ridgeplot" class="figure mx-auto d-block"><div class="figcaption">ridgeplot</div>
</figure><blockquote>
<h2 id="challenge-6">Challenge 6</h2>
<p>How could you plot the results from the ORA to show biodomain
enrichements?</p>
<blockquote>
<h2 id="solution-to-challenge-6">Solution to Challenge 6</h2>
<p>enr.ora = bind_rows( <a href="mailto:enr.up@result" class="email">enr.up@result</a> %&gt;% mutate(dir = ‘up’), <a href="mailto:enr.dn@result" class="email">enr.dn@result</a> %&gt;%
mutate(dir = ‘dn’) ) %&gt;% left_join(., biodom %&gt;% select(Biodomain,
ID = GO_ID), by = ‘ID’) %&gt;% relocate(Biodomain)</p>
<p>enr.ora<span class="math inline">\(Biodomain[is.na(enr.ora\)</span>Biodomain)] &lt;-
‘none’</p>
<p>bd.tally = tibble(domain = c(unique(biodom<span class="math inline">\(Biodomain), 'none')) %&gt;%
  rowwise() %&gt;%
  mutate(
    n_term = biodom\)</span>GO_ID[ biodom<span class="math inline">\(Biodomain == domain ] %&gt;% unique() %&gt;%
length(),
    n_sig_term = enr.ora\)</span>ID[ enr.ora$Biodomain == domain ]
%&gt;% unique() %&gt;% length() )</p>
<p>enr.ora &lt;- full_join(enr.ora, dom.lab, by = c(‘Biodomain’ =
‘domain’)) %&gt;% mutate(Biodomain = factor(Biodomain, levels =
arrange(bd.tally, n_sig_term) %&gt;% pull(domain))) %&gt;%
arrange(Biodomain, p.adjust) %&gt;% mutate(signed_logP =
-log10(p.adjust), signed_logP = if_else(dir == ‘dn’, -1*signed_logP,
signed_logP))</p>
<p>ggplot(enr.ora, aes(signed_logP, Biodomain)) + geom_violin(data =
subset(enr.ora, dir == ‘up’), aes(color = color), scale = ‘width’)+
geom_violin(data = subset(enr.ora, dir == ‘dn’), aes(color = color),
scale = ‘width’)+ geom_jitter(aes(size = Count, fill = color), color =
‘grey20’, shape = 21, alpha = .3)+ geom_vline(xintercept = 0, lwd = .1)+
scale_y_discrete(drop = F)+ scale_fill_identity()+scale_color_identity()
~~~ {: .language-r} Based on the gene list (up or down) we can add a
sign to the significance. When we plot this we can see there are many
more significantly enriched terms from the ORA. The dominant signal is
still the up-regulation of terms from the <code>Immune Response</code>
biodomain. There is also nearly exclusive up-regulation of terms from
the <code>Autophagy</code>, <code>Metal Binding and Homeostasis</code>,
<code>Oxidative Stress</code>, and <code>APP Metabolism</code> domains.
The most down-regulated terms are from the <code>Synapse</code>
biodomain. {: .solution} {: .challenge}</p>
</blockquote>
</blockquote>
</div>
<div class="section level3">
<h3 id="comparing-human-and-mouse-functional-enrichments">Comparing Human and Mouse functional enrichments<a class="anchor" aria-label="anchor" href="#comparing-human-and-mouse-functional-enrichments"></a>
</h3>
<p>Finally let’s consider how we can compare the functional enrichments
in the mouse models with functional enrichments from the AMP-AD cohorts.
We can first take a look at the functional enrichments from the
different human cohorts. To save some time, let’s load some pre-computed
enrichment results for mouse &amp; human data.</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mm.gsea</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span></span>
<span>  <span class="fu">read_tsv</span><span class="op">(</span><span class="st">'data/5xFAD_gsea_results.tsv'</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>age <span class="op">=</span> <span class="fu">str_remove_all</span><span class="op">(</span><span class="va">age</span>, <span class="st">' mo'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">as.double</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  <span class="fu">read_tsv</span><span class="op">(</span><span class="st">'data/LOAD1_gsea_results.tsv'</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in bind_rows(read_tsv("data/5xFAD_gsea_results.tsv", col_types = cols()) %&gt;% : could not find function "bind_rows"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.gsea</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="st">'data/Hsap_gsea_results.tsv'</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in read_tsv("data/Hsap_gsea_results.tsv", col_types = cols()): could not find function "read_tsv"</code></pre>
</div>
<p>The cohorts and tissues available are:</p>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in hs.gsea %&gt;% select(Study, Tissue) %&gt;% distinct(): could not find function "%&gt;%"</code></pre>
</div>
<p>We can focus on one tissue per cohort, let’s look at TCX for Mayo,
PHG for Mt Sinai, and DLPFC for ROSMAP.</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd</span> <span class="op">=</span> <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'TCX'</span>,<span class="st">'PHG'</span>,<span class="st">'DLPFC'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, ID <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'ID'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span> <span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span> <span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">enr.bd</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span> <span class="va">enr.bd</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">enr.bd</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Study</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="va">n_sig_term</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">p.adjust</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.adjust</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span>, group <span class="op">=</span> <span class="va">ID</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.5</span>, position <span class="op">=</span> <span class="fu">position_jitter</span><span class="op">(</span>width <span class="op">=</span> <span class="fl">.25</span>, seed <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>  <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="fu">paste0</span><span class="op">(</span><span class="va">Study</span>, <span class="st">': '</span>, <span class="va">Tissue</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Huh, the Mayo and MSSM look largely similar, but the ROSMAP DLPFC
doesn’t have very many enriched terms. Let’s ignore the ROSMAP result
for now and compare the terms enriched between Mayo TCX and the 5xFAD
males. First we can join up the enriched terms between the sets into one
data frame.</p>
<div class="codewrapper sourceCode" id="cb76">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comb_5x</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span></span>
<span>  <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">Study</span> <span class="op">==</span> <span class="st">'MAYO'</span>, <span class="va">Tissue</span> <span class="op">==</span> <span class="st">'TCX'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>species <span class="op">=</span> <span class="st">'hs'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="va">ID</span>, <span class="va">Description</span>, <span class="va">species</span>, <span class="va">NES</span>, <span class="va">p.adjust</span><span class="op">)</span>,</span>
<span>  <span class="va">mm.gsea</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">model</span> <span class="op">==</span> <span class="st">'5xFAD'</span>, <span class="va">sex</span> <span class="op">==</span> <span class="st">'male'</span>, <span class="va">age</span> <span class="op">==</span> <span class="fl">12</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>species <span class="op">=</span> <span class="st">'mm'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="va">ID</span>, <span class="va">Description</span>, <span class="va">species</span>, <span class="va">NES</span>, <span class="va">p.adjust</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">ID</span>, <span class="va">Description</span><span class="op">)</span>, </span>
<span>              names_from <span class="op">=</span> <span class="va">species</span>, </span>
<span>              values_from <span class="op">=</span> <span class="va">NES</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unnest</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">hs</span>,<span class="va">mm</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">c</span><span class="op">(</span><span class="va">hs</span>,<span class="va">mm</span><span class="op">)</span>, <span class="op">~</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">0</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in bind_rows(hs.gsea %&gt;% filter(Study == "MAYO", Tissue == "TCX") %&gt;% : could not find function "%&gt;%"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb78">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">comb_5x</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'comb_5x' not found</code></pre>
</div>
<p>Great! Now let’s annotate the biodomains and plot it up:</p>
<div class="codewrapper sourceCode" id="cb80">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd</span> <span class="op">&lt;-</span> <span class="va">comb_5x</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, ID <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'ID'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span> <span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span> <span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">enr.bd</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span> <span class="va">enr.bd</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">enr.bd</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, </span>
<span>                            levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">n_sig_term</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">hs</span>, <span class="va">mm</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'Mayo TCX\nGSEA NES'</span>, y <span class="op">=</span> <span class="st">'12mo male 5xFAD\nGSEA NES'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">Biodomain</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>There are several terms from multiple biodomains that are enriched in
similar directions (i.e. from the up-regulated genes in both human AD
and 5xFAD). There are fewer that are enriched in opposite directions
(i.e. up in human AD and down in 5xFAD, e.g. see Vasculature and
Mitochondrial Metabolism), but there are many terms that are unique to
either the human or the mouse data set.</p>
<p>Overall, by aligning human and mouse omics signatures through the
lens of domains affected in each context, we can get a better
understanding of the relationships between the biological processes
affected in each context.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul>
<li>AMP-AD gene modules represent major transcriptomic heterogeneity in
AD</li>
<li>Correlation of logFC is a practical approach for human-mouse
alignment of AD-associated transcriptomic signatures</li>
<li>Functional gene set signatures are also a useful point of comparison
between species</li>
<li>Subsetting functions into associated Biological Domains allows for
more granular comparisons.</li>
</ul>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb81-2"><a href="#cb81-2" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">4.1</span> (<span class="dv">2024-06-14</span>)</span>
<span id="cb81-3"><a href="#cb81-3" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>pc<span class="sc">-</span>linux<span class="sc">-</span>gnu</span>
<span id="cb81-4"><a href="#cb81-4" tabindex="-1"></a>Running under<span class="sc">:</span> Ubuntu <span class="dv">22</span>.<span class="fl">04.4</span> LTS</span>
<span id="cb81-5"><a href="#cb81-5" tabindex="-1"></a></span>
<span id="cb81-6"><a href="#cb81-6" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb81-7"><a href="#cb81-7" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>blas<span class="sc">/</span>libblas.so.<span class="dv">3</span>.<span class="fl">10.0</span> </span>
<span id="cb81-8"><a href="#cb81-8" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>lapack<span class="sc">/</span>liblapack.so.<span class="dv">3</span>.<span class="fl">10.0</span></span>
<span id="cb81-9"><a href="#cb81-9" tabindex="-1"></a></span>
<span id="cb81-10"><a href="#cb81-10" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb81-11"><a href="#cb81-11" tabindex="-1"></a> [<span class="dv">1</span>] LC_CTYPE<span class="ot">=</span>C.UTF<span class="dv">-8</span>       LC_NUMERIC<span class="ot">=</span>C           LC_TIME<span class="ot">=</span>C.UTF<span class="dv">-8</span>       </span>
<span id="cb81-12"><a href="#cb81-12" tabindex="-1"></a> [<span class="dv">4</span>] LC_COLLATE<span class="ot">=</span>C.UTF<span class="dv">-8</span>     LC_MONETARY<span class="ot">=</span>C.UTF<span class="dv">-8</span>    LC_MESSAGES<span class="ot">=</span>C.UTF<span class="dv">-8</span>   </span>
<span id="cb81-13"><a href="#cb81-13" tabindex="-1"></a> [<span class="dv">7</span>] LC_PAPER<span class="ot">=</span>C.UTF<span class="dv">-8</span>       LC_NAME<span class="ot">=</span>C              LC_ADDRESS<span class="ot">=</span>C          </span>
<span id="cb81-14"><a href="#cb81-14" tabindex="-1"></a>[<span class="dv">10</span>] LC_TELEPHONE<span class="ot">=</span>C         LC_MEASUREMENT<span class="ot">=</span>C.UTF<span class="dv">-8</span> LC_IDENTIFICATION<span class="ot">=</span>C   </span>
<span id="cb81-15"><a href="#cb81-15" tabindex="-1"></a></span>
<span id="cb81-16"><a href="#cb81-16" tabindex="-1"></a>time zone<span class="sc">:</span> UTC</span>
<span id="cb81-17"><a href="#cb81-17" tabindex="-1"></a>tzcode source<span class="sc">:</span> <span class="fu">system</span> (glibc)</span>
<span id="cb81-18"><a href="#cb81-18" tabindex="-1"></a></span>
<span id="cb81-19"><a href="#cb81-19" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb81-20"><a href="#cb81-20" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb81-21"><a href="#cb81-21" tabindex="-1"></a></span>
<span id="cb81-22"><a href="#cb81-22" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb81-23"><a href="#cb81-23" tabindex="-1"></a>[<span class="dv">1</span>] compiler_4.<span class="fl">4.1</span>  tools_4.<span class="fl">4.1</span>     yaml_2.<span class="fl">3.9</span>      knitr_1<span class="fl">.48</span>     </span>
<span id="cb81-24"><a href="#cb81-24" tabindex="-1"></a>[<span class="dv">5</span>] xfun_0<span class="fl">.46</span>       renv_1.<span class="fl">0.7</span>      evaluate_0.<span class="fl">24.0</span></span></code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-md/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-md/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-md/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:team@carpentries.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.5" class="external-link">sandpaper (0.16.5)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.3" class="external-link">varnish (1.0.3)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-md/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "Alzheimers, data, lesson, MODEL-AD",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-md/instructor/aio.html",
  "identifier": "https://carpentries.github.io/workbench-template-md/instructor/aio.html",
  "dateCreated": "2024-07-24",
  "dateModified": "2024-07-25",
  "datePublished": "2024-07-25"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

