<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Omics Translation in AD: Synapse and AD Knowledge Portal</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="manifest" href="../site.webmanifest"><link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav "><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="-logo" alt="" src="../assets/images/-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../exploring-data-in-synapse.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav " aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="" src="../assets/images/-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Omics Translation in AD
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Omics Translation in AD
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Omics Translation in AD
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 5%" class="percentage">
    5%
  </div>
  <div class="progress ">
    <div class="progress-bar " role="progressbar" style="width: 5%" aria-valuenow="5" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../exploring-data-in-synapse.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction to Omics Translation in AD</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        2. Synapse and AD Knowledge Portal
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#working-with-ad-portal-metadata">Working with AD Portal metadata</a></li>
<li><a href="#single-specimen-files">Single Specimen files</a></li>
<li><a href="#annotations-during-bulk-download">Annotations during bulk download</a></li>
<li><a href="#multispecimen-files">Multispecimen files</a></li>
<li><a href="#session-info">Session Info</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="differential-expression-analysis.html">3. Differential expression analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="mouse-human-alignment.html">4. Mouse-human alignment of transcriptomic signatures</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/introduction.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-analysis.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/introduction.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Introduction to
        </a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-analysis.html" rel="next">
          Next: Differential...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Synapse and AD Knowledge Portal</h1>
        <p>Last updated on 2024-08-27 |

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/episodes/exploring-data-in-synapse.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How to work with Synapse R client?</li>
<li>How to work with data in AD Knowledge Portal?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Explain how to use Synapser Package.</li>
<li>Demonstrate how to locate data and metadata in the Portal.</li>
<li>Demonstrate how to download data from the Portal
programmatically.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="working-with-ad-portal-metadata">Working with AD Portal metadata<a class="anchor" aria-label="anchor" href="#working-with-ad-portal-metadata"></a></h2>
<hr class="half-width"><p><strong>Metadata basics</strong></p>
<p>We have now downloaded several metadata files and an RNAseq counts
file from the portal. For our next exercises, we want to read those
files in as R data so we can work with them.</p>
<p>We can see from the download_table we got during the bulk download
step that we have five metadata files. Two of these should be the
individual and biospecimen files, and three of them are assay metadata
files.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">download_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">metadataType</span>, <span class="va">assay</span><span class="op">)</span></span></code></pre>
</div>
<p>We are only interested in RNAseq data, so we will only read in the
individual, biospecimen, and RNAseq assay metadata files.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># counts matrix</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="st">"data/htseqcounts_5XFAD.txt"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_individual_metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_biospecimen_metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#assay metadata</span></span>
<span><span class="va">rna_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_assay_RNAseq_metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s examine the data and metadata files a bit before we begin our
analyses.</p>
<p><strong>Counts data</strong></p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calling a tibble object will print the first ten rows in a nice tidy output; doing the same for a base R dataframe will print the whole thing until it runs out of memory. If you want to inspect a large dataframe, use `head(df)`</span></span>
<span><span class="va">counts</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 55,489 × 73
   gene_id `32043rh` `32044rh` `32046rh` `32047rh` `32048rh` `32049rh` `32050rh`
   &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1 ENSG00…     22554         0         0         0     16700         0         0
 2 ENSG00…    344489         4         0         1    260935         6         8
 3 ENSMUS…      5061      3483      3941      3088      2756      3067      2711
 4 ENSMUS…         0         0         0         0         0         0         0
 5 ENSMUS…       208       162       138       127        95       154       165
 6 ENSMUS…        44        17        14        28        23        24        14
 7 ENSMUS…       143        88       121       117       115       109        75
 8 ENSMUS…        22         6        10        11        11        19        24
 9 ENSMUS…      7165      5013      5581      4011      4104      5254      4345
10 ENSMUS…      3728      2316      2238      1965      1822      1999      1809
# ℹ 55,479 more rows
# ℹ 65 more variables: `32052rh` &lt;dbl&gt;, `32053rh` &lt;dbl&gt;, `32057rh` &lt;dbl&gt;,
#   `32059rh` &lt;dbl&gt;, `32061rh` &lt;dbl&gt;, `32062rh` &lt;dbl&gt;, `32065rh` &lt;dbl&gt;,
#   `32067rh` &lt;dbl&gt;, `32068rh` &lt;dbl&gt;, `32070rh` &lt;dbl&gt;, `32073rh` &lt;dbl&gt;,
#   `32074rh` &lt;dbl&gt;, `32075rh` &lt;dbl&gt;, `32078rh` &lt;dbl&gt;, `32081rh` &lt;dbl&gt;,
#   `32088rh` &lt;dbl&gt;, `32640rh` &lt;dbl&gt;, `46105rh` &lt;dbl&gt;, `46106rh` &lt;dbl&gt;,
#   `46107rh` &lt;dbl&gt;, `46108rh` &lt;dbl&gt;, `46109rh` &lt;dbl&gt;, `46110rh` &lt;dbl&gt;, …</code></pre>
</div>
<p>The data file has a column of ENSEMBL gene ids and then a bunch of
columns with count data, where the column headers correspond to the
specimenIDs. These specimenIDs should all be in the RNAseq assay
metadata file, so let’s check.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># what does the RNAseq assay metadata look like?</span></span>
<span><span class="va">rna_meta</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 72 × 12
   specimenID platform   RIN   rnaBatch libraryBatch sequencingBatch libraryPrep
   &lt;chr&gt;      &lt;chr&gt;      &lt;lgl&gt;    &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;
 1 32043rh    IlluminaN… NA           1            1               1 polyAselec…
 2 32044rh    IlluminaN… NA           1            1               1 polyAselec…
 3 32046rh    IlluminaN… NA           1            1               1 polyAselec…
 4 32047rh    IlluminaN… NA           1            1               1 polyAselec…
 5 32049rh    IlluminaN… NA           1            1               1 polyAselec…
 6 32057rh    IlluminaN… NA           1            1               1 polyAselec…
 7 32061rh    IlluminaN… NA           1            1               1 polyAselec…
 8 32065rh    IlluminaN… NA           1            1               1 polyAselec…
 9 32067rh    IlluminaN… NA           1            1               1 polyAselec…
10 32070rh    IlluminaN… NA           1            1               1 polyAselec…
# ℹ 62 more rows
# ℹ 5 more variables: libraryPreparationMethod &lt;lgl&gt;, isStranded &lt;lgl&gt;,
#   readStrandOrigin &lt;lgl&gt;, runType &lt;chr&gt;, readLength &lt;dbl&gt;</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># are all the column headers from the counts matrix (except the first "gene_id" column) in the assay metadata?</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<p><strong>Assay metadata</strong></p>
<p>The assay metadata contains information about how data was generated
on each sample in the assay. Each specimenID represents a unique sample.
We can use some tools from dplyr to explore the metadata.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># how many unique specimens were sequenced?</span></span>
<span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 72</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># were the samples all sequenced on the same platform?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">rna_meta</span>, <span class="va">platform</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 1
  platform
  &lt;chr&gt;
1 IlluminaNovaseq6000</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># were there multiple sequencing batches reported?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">rna_meta</span>, <span class="va">sequencingBatch</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 1
  sequencingBatch
            &lt;dbl&gt;
1               1</code></pre>
</div>
<p><strong>Biospecimen metadata</strong></p>
<p>The biospecimen metadata contains specimen-level information,
including organ and tissue the specimen was taken from, how it was
prepared, etc. Each specimenID is mapped to an individualID.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all specimens from the RNAseq assay metadata file should be in the biospecimen file</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span> <span class="op">%in%</span> <span class="va">bio_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># but the biospecimen file also contains specimens from different assays</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">bio_meta</span><span class="op">$</span><span class="va">specimenID</span> <span class="op">%in%</span> <span class="va">rna_meta</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] FALSE</code></pre>
</div>
<p><strong>Individual metadata</strong></p>
<p>The individual metadata contains information about all the
individuals in the study, represented by unique individualIDs. For
humans, this includes information on age, sex, race, diagnosis, etc. For
MODEL-AD mouse models, the individual metadata has information on model
genotypes, stock numbers, diet, and more.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all individualIDs in the biospecimen file should be in the individual file</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">bio_meta</span><span class="op">$</span><span class="va">individualID</span> <span class="op">%in%</span> <span class="va">ind_meta</span><span class="op">$</span><span class="va">individualID</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># which model genotypes are in this study?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">ind_meta</span>, <span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 2 × 1
  genotype
  &lt;chr&gt;
1 5XFAD_carrier
2 5XFAD_noncarrier</code></pre>
</div>
<p><strong>Joining metadata</strong></p>
<p>We use the three-file structure for our metadata because it allows us
to store metadata for each study in a tidy format. Every line in the
assay and biospecimen files represents a unique specimen, and every line
in the individual file represents a unique individual. This means the
files can be easily joined by specimenID and individualID to get all
levels of metadata that apply to a particular data file. We will use the
left_join() function from the dplyr package, and the %&gt;% operator
from the magrittr package. If you are unfamiliar with the pipe, think of
it as a shorthand for “take this (the preceding object) and do that (the
subsequent command)”. See here (<a href="https://magrittr.tidyverse.org/" class="external-link uri">https://magrittr.tidyverse.org/</a>) for more info on piping
in R.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># join all the rows in the assay metadata that have a match in the biospecimen metadata</span></span>
<span><span class="va">joined_meta</span> <span class="op">&lt;-</span> <span class="va">rna_meta</span> <span class="op">%&gt;%</span> <span class="co">#start with the rnaseq assay metadata</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">bio_meta</span>, by <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span>  <span class="co">#join rows from biospecimen that match specimenID </span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">ind_meta</span>, by <span class="op">=</span> <span class="st">"individualID"</span><span class="op">)</span> <span class="co"># join rows from individual that match individualID</span></span>
<span></span>
<span><span class="va">joined_meta</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 72 × 53
   specimenID platform   RIN   rnaBatch libraryBatch sequencingBatch libraryPrep
   &lt;chr&gt;      &lt;chr&gt;      &lt;lgl&gt;    &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;
 1 32043rh    IlluminaN… NA           1            1               1 polyAselec…
 2 32044rh    IlluminaN… NA           1            1               1 polyAselec…
 3 32046rh    IlluminaN… NA           1            1               1 polyAselec…
 4 32047rh    IlluminaN… NA           1            1               1 polyAselec…
 5 32049rh    IlluminaN… NA           1            1               1 polyAselec…
 6 32057rh    IlluminaN… NA           1            1               1 polyAselec…
 7 32061rh    IlluminaN… NA           1            1               1 polyAselec…
 8 32065rh    IlluminaN… NA           1            1               1 polyAselec…
 9 32067rh    IlluminaN… NA           1            1               1 polyAselec…
10 32070rh    IlluminaN… NA           1            1               1 polyAselec…
# ℹ 62 more rows
# ℹ 46 more variables: libraryPreparationMethod &lt;lgl&gt;, isStranded &lt;lgl&gt;,
#   readStrandOrigin &lt;lgl&gt;, runType &lt;chr&gt;, readLength &lt;dbl&gt;,
#   individualID &lt;dbl&gt;, specimenIdSource &lt;chr&gt;, organ &lt;chr&gt;, tissue &lt;chr&gt;,
#   BrodmannArea &lt;lgl&gt;, sampleStatus &lt;chr&gt;, tissueWeight &lt;lgl&gt;,
#   tissueVolume &lt;lgl&gt;, nucleicAcidSource &lt;lgl&gt;, cellType &lt;lgl&gt;,
#   fastingState &lt;lgl&gt;, isPostMortem &lt;lgl&gt;, samplingAge &lt;lgl&gt;, …</code></pre>
</div>
<p>We now have a very wide dataframe that contains all the available
metadata on each specimen in the RNAseq data from this study. This
procedure can be used to join the three types of metadata files for
every study in the AD Knowledge Portal, allowing you to filter
individuals and specimens as needed based on your analysis criteria!</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">lubridate</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># convert columns of strings to month-date-year format</span></span>
<span><span class="va">joined_meta_time</span> <span class="op">&lt;-</span> <span class="va">joined_meta</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>dateBirth <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateBirth</span><span class="op">)</span>, dateDeath <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateDeath</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="co"># create a new column that subtracts dateBirth from dateDeath in days, then divide by 30 to get months</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>timepoint <span class="op">=</span> <span class="fu">as.numeric</span><span class="op">(</span><span class="fu">difftime</span><span class="op">(</span><span class="va">dateDeath</span>, <span class="va">dateBirth</span>, units <span class="op">=</span><span class="st">"days"</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fl">30</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="co"># convert numeric ages to timepoint categories</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>timepoint <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span><span class="va">timepoint</span> <span class="op">&gt;</span> <span class="fl">10</span> <span class="op">~</span> <span class="st">"12 mo"</span>,</span>
<span>                               <span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">10</span> <span class="op">&amp;</span> <span class="va">timepoint</span> <span class="op">&gt;</span> <span class="fl">5</span> <span class="op">~</span> <span class="st">"6 mo"</span>,</span>
<span>                               <span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">5</span> <span class="op">~</span> <span class="st">"4 mo"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covars_5XFAD</span> <span class="op">&lt;-</span> <span class="va">joined_meta_time</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">individualID</span>, <span class="va">specimenID</span>, <span class="va">sex</span>, <span class="va">genotype</span>, <span class="va">timepoint</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">covars_5XFAD</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">covars_5XFAD</span><span class="op">$</span><span class="va">specimenID</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">covars_5XFAD</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        individualID specimenID    sex         genotype timepoint
32043rh        32043    32043rh female    5XFAD_carrier     12 mo
32044rh        32044    32044rh   male 5XFAD_noncarrier     12 mo
32046rh        32046    32046rh   male 5XFAD_noncarrier     12 mo
32047rh        32047    32047rh   male 5XFAD_noncarrier     12 mo
32049rh        32049    32049rh female 5XFAD_noncarrier     12 mo
32057rh        32057    32057rh female 5XFAD_noncarrier     12 mo</code></pre>
</div>
<p>We will save joined_meta for the next lesson.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">covars_5XFAD</span>, file <span class="op">=</span> <span class="st">"data/covars_5XFAD.rds"</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="single-specimen-files">Single Specimen files<a class="anchor" aria-label="anchor" href="#single-specimen-files"></a></h2>
<hr class="half-width"><p>For files that contain data from a single specimen (e.g. raw
sequencing files, raw mass spectra, etc.), we can use the Synapse
annotations to associate these files with the appropriate metadata.</p>
<p>Excercise 3: Use Explore Data to find all RNAseq files from the
Jax.IU.Pitt_5XFAD study. If we filter for data where Study =
“Jax.IU.Pitt_5XFAD” and Assay = “rnaSeq” we will get a list of 148
files, including raw fastqs and processed counts data.</p>
<p>Synapse entity annotations We can use the function synGetAnnotations
to view the annotations associated with any file without downloading the
file.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the synID of a random fastq file from this list</span></span>
<span><span class="va">random_fastq</span> <span class="op">&lt;-</span> <span class="st">"syn22108503"</span></span>
<span></span>
<span><span class="co"># extract the annotations as a nested list</span></span>
<span><span class="va">fastq_annotations</span> <span class="op">&lt;-</span> <span class="fu">synGetAnnotations</span><span class="op">(</span><span class="va">random_fastq</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fastq_annotations</span></span></code></pre>
</div>
<p>The file annotations let us see which study the file is associated
with (Jax.IU.Pitt.5XFAD), which species it’s from (Mouse), which assay
generated the file (rnaSeq), and a whole bunch of other properties. Most
importantly, single-specimen files are annotated with with the
specimenID of the specimen in the file, and the individualID of the
individual that specimen was taken from. We can use these annotations to
link files to the rest of the metadata, including metadata that is not
in annotations. This is especially helpful for human studies, as
potentially identifying information like age, race, and diagnosis is not
included in file annotations.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># find records belonging to the individual this file maps to in our joined metadata</span></span>
<span><span class="va">joined_meta</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">individualID</span> <span class="op">==</span> <span class="va">fastq_annotations</span><span class="op">$</span><span class="va">individualID</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="annotations-during-bulk-download">Annotations during bulk download<a class="anchor" aria-label="anchor" href="#annotations-during-bulk-download"></a></h2>
<hr class="half-width"><p>When bulk downloading many files, the best practice is to preserve
the download manifest that is generated which lists all the files, their
synIDs, and all their annotations. If using the Synapse R client, follow
the instructions in the Bulk download files section above.</p>
<p>If we use the “Programmatic Options” tab in the AD Portal download
menu to download all 148 rnaSeq files from the 5XFAD study, we would get
a table query that looks like this:</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">synTableQuery</span><span class="op">(</span><span class="st">"SELECT * FROM syn11346063.37 WHERE ( ( \"study\" HAS ( 'Jax.IU.Pitt_5XFAD' ) ) AND ( \"assay\" HAS ( 'rnaSeq' ) ) )"</span><span class="op">)</span></span></code></pre>
</div>
<p>As we saw previously, this downloads a csv file with the results of
our AD Portal query. Opening that file lets us see which specimens are
associated with which files:</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations_table</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">query</span><span class="op">$</span><span class="va">filepath</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">annotations_table</span></span></code></pre>
</div>
<p>You could then use purrr::walk(download_table$id, ~synGet(.x,
downloadLocation = )) to walk through the column of synIDs and download
all 148 files. However, because these are large files, it might be
preferable to use the Python client or command line client for increased
speed.</p>
<p>Once you’ve downloaded all the files in the id column, you can link
those files to their annotations by the name column.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We'll use the "random fastq" that we got annotations for earlier</span></span>
<span><span class="co"># To avoid downloading the whole 3GB file, we'll use synGet with "downloadFile = FALSE" to get only the Synapse entity information, rather than the file. </span></span>
<span><span class="co"># If we downloaded the actual file, we could find it in the directory and search using the filename. Since we're just downloading the Synapse entity wrapper object, we'll use the file name listed in the object properties.</span></span>
<span></span>
<span><span class="va">fastq</span> <span class="op">&lt;-</span> <span class="fu">synGet</span><span class="op">(</span><span class="va">random_fastq</span>, downloadFile <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the annotations table to rows that match the fastq filename</span></span>
<span><span class="va">annotations_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="va">fastq</span><span class="op">$</span><span class="va">properties</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="multispecimen-files">Multispecimen files<a class="anchor" aria-label="anchor" href="#multispecimen-files"></a></h2>
<hr class="half-width"><p>Multispecimen files in the AD Knowledge Portal are files that contain
data or information from more than one specimen. They are not annotated
with individualIDs or specimenIDs, since these files may contain numbers
of specimens that exceed the annotation limits. These files are usually
processed or summary data (gene counts, peptide quantifications, etc),
and are always annotated with isMultiSpecimen = TRUE.</p>
<p>If we look at the processed data files in the table of 5XFAD RNAseq
file annotations we just downloaded, we will see that it isMultiSpecimen
= TRUE, but individualID and specimenID are blank:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">fileFormat</span> <span class="op">==</span> <span class="st">"txt"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">individualID</span>, <span class="va">specimenID</span>, <span class="va">isMultiSpecimen</span><span class="op">)</span></span></code></pre>
</div>
<p>The multispecimen file should contain a row or column of specimenIDs
that correspond to the specimenIDs used in a study’s metadata, as we
have seen with the 5XFAD counts file.</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># In this example, we take a slice of the counts data to reduce computation, transpose it so that each row represents a single specimen, and then join it to the joined metadata by the specimenID</span></span>
<span><span class="va">counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">slice_head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">t</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span>rownames <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">joined_meta</span>, by <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 73 × 58
   specimenID V1    V2    V3    V4    V5    platform RIN   rnaBatch libraryBatch
   &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;lgl&gt;    &lt;dbl&gt;        &lt;dbl&gt;
 1 gene_id    "ENS… "ENS… "ENS… "ENS… "ENS… &lt;NA&gt;     NA          NA           NA
 2 32043rh    " 22… "344… "  5… "   … "   … Illumin… NA           1            1
 3 32044rh    "   … "   … "348… "   … " 16… Illumin… NA           1            1
 4 32046rh    "   … "   … "394… "   … " 13… Illumin… NA           1            1
 5 32047rh    "   … "   … "308… "   … " 12… Illumin… NA           1            1
 6 32048rh    " 16… "260… "  2… "   … "   … Illumin… NA           1            1
 7 32049rh    "   … "   … "306… "   … " 15… Illumin… NA           1            1
 8 32050rh    "   … "   … "271… "   … " 16… Illumin… NA           1            1
 9 32052rh    " 19… "337… "  3… "   … "   … Illumin… NA           1            1
10 32053rh    " 14… "206… "  3… "   … "   … Illumin… NA           1            1
# ℹ 63 more rows
# ℹ 48 more variables: sequencingBatch &lt;dbl&gt;, libraryPrep &lt;chr&gt;,
#   libraryPreparationMethod &lt;lgl&gt;, isStranded &lt;lgl&gt;, readStrandOrigin &lt;lgl&gt;,
#   runType &lt;chr&gt;, readLength &lt;dbl&gt;, individualID &lt;dbl&gt;,
#   specimenIdSource &lt;chr&gt;, organ &lt;chr&gt;, tissue &lt;chr&gt;, BrodmannArea &lt;lgl&gt;,
#   sampleStatus &lt;chr&gt;, tissueWeight &lt;lgl&gt;, tissueVolume &lt;lgl&gt;,
#   nucleicAcidSource &lt;lgl&gt;, cellType &lt;lgl&gt;, fastingState &lt;lgl&gt;, …</code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Use your Synapse login credentials to access the Portal.</li>
<li>Use Synapser package to download data from the Portal.</li>
</ul></div>
</div>
</div>
</section><section><h2 class="section-heading" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a></h2>
<hr class="half-width"><div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessionInfo</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] lubridate_1.9.3 forcats_1.0.0   stringr_1.5.1   purrr_1.0.2
 [5] readr_2.1.5     tidyr_1.3.1     tibble_3.2.1    ggplot2_3.5.1
 [9] tidyverse_2.0.0 dplyr_1.1.4

loaded via a namespace (and not attached):
 [1] bit_4.0.5        gtable_0.3.5     compiler_4.4.1   renv_1.0.7
 [5] crayon_1.5.3     tidyselect_1.2.1 parallel_4.4.1   scales_1.3.0
 [9] yaml_2.3.10      R6_2.5.1         generics_0.1.3   knitr_1.48
[13] munsell_0.5.1    pillar_1.9.0     tzdb_0.4.0       rlang_1.1.4
[17] utf8_1.2.4       stringi_1.8.4    xfun_0.47        bit64_4.0.5
[21] timechange_0.3.0 cli_3.6.3        withr_3.0.1      magrittr_2.0.3
[25] grid_4.4.1       vroom_1.6.5      hms_1.1.3        lifecycle_1.0.4
[29] vctrs_0.6.5      evaluate_0.24.0  glue_1.7.0       fansi_1.0.6
[33] colorspace_2.1-1 tools_4.4.1      pkgconfig_2.0.3 </code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/introduction.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-analysis.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/introduction.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Introduction to
        </a>
        <a class="chapter-link float-end" href="../instructor/differential-expression-analysis.html" rel="next">
          Next: Differential...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-md/edit/main/episodes/exploring-data-in-synapse.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-md/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-md/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-md/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:team@carpentries.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.6" class="external-link">sandpaper (0.16.6)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.4" class="external-link">varnish (1.0.4)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-md/instructor/exploring-data-in-synapse.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "Alzheimers, data, lesson, MODEL-AD",
  "name": "Synapse and AD Knowledge Portal",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-md/instructor/exploring-data-in-synapse.html",
  "identifier": "https://carpentries.github.io/workbench-template-md/instructor/exploring-data-in-synapse.html",
  "dateCreated": "2024-07-24",
  "dateModified": "2024-08-27",
  "datePublished": "2024-08-27"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

